From d082676932a418c8095ea20b3d43f0e4b387b7f5 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Thu, 17 Feb 2022 23:23:24 -0800
Subject: [PATCH] Handle tparm constants with type const char *

---
 sfeed_curses.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/sfeed_curses.c b/sfeed_curses.c
index 4087369..9bd65d7 100644
--- a/sfeed_curses.c
+++ b/sfeed_curses.c
@@ -267,12 +267,12 @@ estrdup(const char *s)
 
 /* Wrapper for tparm which allows NULL parameter for str. */
 char *
-tparmnull(char *str, long p1, long p2, long p3, long p4, long p5, long p6,
-          long p7, long p8, long p9)
+tparmnull(const char *str, long p1, long p2, long p3, long p4, long p5,
+          long p6, long p7, long p8, long p9)
 {
 	if (!str)
 		return NULL;
-	return tparm(str, p1, p2, p3, p4, p5, p6, p7, p8, p9);
+	return tparm((char *)str, p1, p2, p3, p4, p5, p6, p7, p8, p9);
 }
 
 /* Counts column width of character string. */
-- 
2.34.1

