From 7f16d268831910c1d219b8a4f38b364bba34c710 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Mon, 18 Apr 2016 01:16:12 -0700
Subject: [PATCH] tar: Default to stdin

---
 bin/pax/options.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bin/pax/options.c b/bin/pax/options.c
index cd3f5f6285c..9dd2ea0ce25 100644
--- a/bin/pax/options.c
+++ b/bin/pax/options.c
@@ -968,7 +968,7 @@ tar_options(int argc, char **argv)
 	if ((arcname == NULL) || (*arcname == '\0')) {
 		arcname = getenv("TAPE");
 		if ((arcname == NULL) || (*arcname == '\0'))
-			arcname = _PATH_DEFTAPE;
+			arcname = "-";
 	}
 	if ((arcname[0] == '-') && (arcname[1]== '\0'))
 		arcname = NULL;
-- 
2.49.0

