From 901940cdf7fc13516ff55e81df0b546eb7c74595 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Mon, 18 Apr 2016 01:16:12 -0700
Subject: [PATCH] tar: Default to stdin

---
 bin/pax/options.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bin/pax/options.c b/bin/pax/options.c
index 5db0948858c..dbb4b816c37 100644
--- a/bin/pax/options.c
+++ b/bin/pax/options.c
@@ -937,7 +937,7 @@ tar_options(int argc, char **argv)
 	if ((arcname == NULL) || (*arcname == '\0')) {
 		arcname = getenv("TAPE");
 		if ((arcname == NULL) || (*arcname == '\0'))
-			arcname = _PATH_DEFTAPE;
+			arcname = "-";
 	}
 	if ((arcname[0] == '-') && (arcname[1]== '\0'))
 		arcname = NULL;
-- 
2.25.0

