From f926c3a15c772a7d54561e01682dea486c2a2833 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Mon, 18 Apr 2016 17:34:01 -0700
Subject: [PATCH] yacc: Add some includes

---
 usr.bin/yacc/defs.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/usr.bin/yacc/defs.h b/usr.bin/yacc/defs.h
index 7300caec2..fab6e1266 100644
--- a/usr.bin/yacc/defs.h
+++ b/usr.bin/yacc/defs.h
@@ -35,6 +35,7 @@
  *	@(#)defs.h	5.6 (Berkeley) 5/24/93
  */
 
+#include <sys/cdefs.h>
 #include <assert.h>
 #include <ctype.h>
 #include <stdio.h>
-- 
2.12.2

