From 299d570c708b33349131b8ca925c3e9bb733db84 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sat, 4 Jun 2016 18:40:24 -0700
Subject: [PATCH] patch: Add missing includes

---
 usr.bin/patch/patch.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/usr.bin/patch/patch.c b/usr.bin/patch/patch.c
index 0cb924db733..17b3aa4e23a 100644
--- a/usr.bin/patch/patch.c
+++ b/usr.bin/patch/patch.c
@@ -26,6 +26,7 @@
  * behaviour
  */
 
+#include <sys/cdefs.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <unistd.h>
-- 
2.19.0

