From 91483dfca3aa6373451120db45c60198faa0766e Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sat, 4 Jun 2016 18:40:37 -0700
Subject: [PATCH] patch: Avoid d_namlen

---
 usr.bin/patch/backupfile.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/usr.bin/patch/backupfile.c b/usr.bin/patch/backupfile.c
index d9e40bcba..1d7858ae3 100644
--- a/usr.bin/patch/backupfile.c
+++ b/usr.bin/patch/backupfile.c
@@ -95,7 +95,7 @@ max_backup_version(const char *file, const char *dir)
 	file_name_length = strlen(file);
 
 	while ((dp = readdir(dirp)) != NULL) {
-		if (dp->d_namlen <= file_name_length)
+		if (strlen(dp->d_name) <= file_name_length)
 			continue;
 
 		this_version = version_number(file, dp->d_name, file_name_length);
-- 
2.12.2

