From 6e087cd1f746eb636de0ec1769db6b4aac65e666 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Fri, 14 Jun 2019 12:42:15 -0700
Subject: [PATCH] rsync: Use standard S_ISVTX instead of S_ISTXT

---
 usr.bin/rsync/receiver.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/usr.bin/rsync/receiver.c b/usr.bin/rsync/receiver.c
index 256a788b4f0..5dcc33b20fd 100644
--- a/usr.bin/rsync/receiver.c
+++ b/usr.bin/rsync/receiver.c
@@ -87,7 +87,7 @@ rsync_set_metadata(struct sess *sess, int newfile,
 				    "to user.group: %u.%u", f->path, uid, gid);
 		} else
 			LOG4("%s: updated uid and/or gid", f->path);
-		mode &= ~(S_ISTXT | S_ISUID | S_ISGID);
+		mode &= ~(S_ISVTX | S_ISUID | S_ISGID);
 	}
 
 	/* Conditionally adjust file permissions. */
@@ -149,7 +149,7 @@ rsync_set_metadata_at(struct sess *sess, int newfile, int rootfd,
 				    "to user.group: %u.%u", f->path, uid, gid);
 		} else
 			LOG4("%s: updated uid and/or gid", f->path);
-		mode &= ~(S_ISTXT | S_ISUID | S_ISGID);
+		mode &= ~(S_ISVTX | S_ISUID | S_ISGID);
 	}
 
 	/* Conditionally adjust file permissions. */
-- 
2.49.0

