From 588522657724047a2e4c838e62dfff5db41bd506 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Fri, 14 Jun 2019 12:44:02 -0700
Subject: [PATCH] rsync: Use -1 instead of INFTIM for poll timeout

---
 usr.bin/rsync/extern.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/usr.bin/rsync/extern.h b/usr.bin/rsync/extern.h
index 040588003a4..64a02349adc 100644
--- a/usr.bin/rsync/extern.h
+++ b/usr.bin/rsync/extern.h
@@ -47,7 +47,7 @@
  * Use this for debugging deadlocks.
  * All poll events will use it and catch time-outs.
  */
-#define POLL_TIMEOUT	(INFTIM)
+#define POLL_TIMEOUT	(-1)
 
 /*
  * Operating mode for a client or a server.
-- 
2.22.0

