From 9121d31f3ca0d49395846334752b6e820da66cf4 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Wed, 15 Apr 2020 20:26:16 -0700
Subject: [PATCH] acme-client: Fix build with old bison versions

---
 usr.sbin/acme-client/parse.y | 1 +
 1 file changed, 1 insertion(+)

diff --git a/usr.sbin/acme-client/parse.y b/usr.sbin/acme-client/parse.y
index 5b9795f948b..14a3cec8203 100644
--- a/usr.sbin/acme-client/parse.y
+++ b/usr.sbin/acme-client/parse.y
@@ -100,6 +100,7 @@ typedef struct {
 	} v;
 	int lineno;
 } YYSTYPE;
+#define YYSTYPE_IS_DECLARED 1
 
 %}
 
-- 
2.54.0

