From e5c8e801a42459bf0c416f6751b687d61f46fcf1 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Thu, 4 Jun 2020 21:44:26 -0700
Subject: [PATCH] pax: Use memcpy to set TMAGIC and TVERSION to avoid warning

---
 bin/pax/tar.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/bin/pax/tar.c b/bin/pax/tar.c
index 8d6b3f37012..1f4012123c3 100644
--- a/bin/pax/tar.c
+++ b/bin/pax/tar.c
@@ -1040,8 +1040,8 @@ ustar_wr(ARCHD *arcn)
 		break;
 	}
 
-	strncpy(hd->magic, TMAGIC, TMAGLEN);
-	strncpy(hd->version, TVERSION, TVERSLEN);
+	memcpy(hd->magic, TMAGIC, TMAGLEN);
+	memcpy(hd->version, TVERSION, TVERSLEN);
 
 	/*
 	 * set the remaining fields. Some versions want all 16 bits of mode
-- 
2.35.1

