From c3aab43559f17feb64a29f8b0d395532cb423e8b Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Wed, 15 Apr 2020 20:26:16 -0700
Subject: [PATCH] acme-client: Fix build with old bison versions

---
 usr.sbin/acme-client/parse.y | 1 +
 1 file changed, 1 insertion(+)

diff --git a/usr.sbin/acme-client/parse.y b/usr.sbin/acme-client/parse.y
index 20818328d92..f392e516b63 100644
--- a/usr.sbin/acme-client/parse.y
+++ b/usr.sbin/acme-client/parse.y
@@ -97,6 +97,7 @@ typedef struct {
 	} v;
 	int lineno;
 } YYSTYPE;
+#define YYSTYPE_IS_DECLARED 1
 
 %}
 
-- 
2.26.1

