From 8cb378367350e0e54688b279e9b99943c192c029 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Wed, 3 Jul 2019 02:20:12 -0700
Subject: [PATCH] video/out/bitmap_packer: Avoid empty initializer list

---
 video/out/bitmap_packer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/video/out/bitmap_packer.c b/video/out/bitmap_packer.c
index 5169357e3a..921b7331ce 100644
--- a/video/out/bitmap_packer.c
+++ b/video/out/bitmap_packer.c
@@ -93,7 +93,7 @@ static int pack_rectangles(struct pos *in, struct pos *out, int num_rects,
         bins[i] = bins[i << HEIGHT_SORT_BITS] - sizes[i << HEIGHT_SORT_BITS];
     struct {
         int size, x, bottom;
-    } stack[16] = {{15, 0, h}}, s = {};
+    } stack[16] = {{15, 0, h}}, s = {0};
     int stackpos = 1;
     int y;
     while (stackpos) {
-- 
2.22.0

