From 7c2361dbc8ceabc1b3edf9e585ccd4e80dd83ea8 Mon Sep 17 00:00:00 2001
From: sfan5 <sfan5@live.de>
Date: Sat, 30 Oct 2021 17:11:46 +0200
Subject: [PATCH] audio: replace deprecated av_mallocz_array

---
 audio/aframe.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/audio/aframe.c b/audio/aframe.c
index c2c0df7c9b..46264b692e 100644
--- a/audio/aframe.c
+++ b/audio/aframe.c
@@ -637,7 +637,7 @@ int mp_aframe_pool_allocate(struct mp_aframe_pool *pool, struct mp_aframe *frame
         av_freep(&av_frame->extended_data); // sigh
     if (planes > AV_NUM_DATA_POINTERS) {
         av_frame->extended_data =
-            av_mallocz_array(planes, sizeof(av_frame->extended_data[0]));
+            av_calloc(planes, sizeof(av_frame->extended_data[0]));
         if (!av_frame->extended_data)
             abort();
     } else {
-- 
2.37.3

