From b8324473e35fbcfec302955456f545572521872b Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sun, 5 Jun 2016 17:28:40 -0700
Subject: [PATCH] Use bool instead of gboolean

---
 sshfs.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sshfs.c b/sshfs.c
index 32db6f4..f43ae5a 100644
--- a/sshfs.c
+++ b/sshfs.c
@@ -16,6 +16,7 @@
 #  include <fuse_darwin.h>
 #endif
 #include <assert.h>
+#include <stdbool.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
@@ -1975,7 +1976,7 @@ static int sftp_request_send(uint8_t type, struct iovec *iov, size_t count,
 
 	err = -EIO;
 	if (sftp_send_iov(type, req->id, iov, count) == -1) {
-		gboolean rmed;
+		bool rmed;
 
 		pthread_mutex_lock(&sshfs.lock);
 		rmed = !!request_table_lookup(&sshfs.reqtab, req->id);
-- 
2.9.0

