From 8e035c27bf466534db0b9233cb038973b1f9ffc5 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Wed, 19 Jun 2019 20:06:29 -0700
Subject: [PATCH] Include stdlib.h for arc4random_uniform

---
 openbsd-compat/bindresvport.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/openbsd-compat/bindresvport.c b/openbsd-compat/bindresvport.c
index eeb269d5..346c7fe5 100644
--- a/openbsd-compat/bindresvport.c
+++ b/openbsd-compat/bindresvport.c
@@ -40,6 +40,7 @@
 #include <arpa/inet.h>
 
 #include <errno.h>
+#include <stdlib.h>
 #include <string.h>
 
 #define STARTPORT 600
-- 
2.20.1

