From fc95498df16315eac1ebe62db51ea33b96ce6b52 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sun, 3 Nov 2019 11:24:13 -0800
Subject: [PATCH] Allow security key provider to be built-in

---
 ssh-sk.c | 30 ++++++++++++++++++------------
 1 file changed, 18 insertions(+), 12 deletions(-)

diff --git a/ssh-sk.c b/ssh-sk.c
index 122a1e2b..ca16a187 100644
--- a/ssh-sk.c
+++ b/ssh-sk.c
@@ -84,16 +84,22 @@ sshsk_open(const char *path)
 		error("%s: strdup failed", __func__);
 		goto fail;
 	}
-	if ((ret->dlhandle = dlopen(path, RTLD_NOW)) == NULL) {
-		error("Security key provider %s dlopen failed: %s",
-		    path, dlerror());
-		goto fail;
-	}
-	if ((ret->sk_api_version = dlsym(ret->dlhandle,
-	    "sk_api_version")) == NULL) {
-		error("Security key provider %s dlsym(sk_api_version) "
-		    "failed: %s", path, dlerror());
-		goto fail;
+	if (strcmp(path, "builtin") == 0) {
+		ret->sk_api_version = sk_api_version;
+		ret->sk_enroll = sk_enroll;
+		ret->sk_sign = sk_sign;
+	} else {
+		if ((ret->dlhandle = dlopen(path, RTLD_NOW)) == NULL) {
+			error("Security key provider %s dlopen failed: %s",
+			    path, dlerror());
+			goto fail;
+		}
+		if ((ret->sk_api_version = dlsym(ret->dlhandle,
+		    "sk_api_version")) == NULL) {
+			error("Security key provider %s dlsym(sk_api_version) "
+			    "failed: %s", path, dlerror());
+			goto fail;
+		}
 	}
 	version = ret->sk_api_version();
 	debug("%s: provider %s implements version 0x%08lx", __func__,
@@ -104,12 +110,12 @@ sshsk_open(const char *path)
 		    (u_long)SSH_SK_VERSION_MAJOR);
 		goto fail;
 	}
-	if ((ret->sk_enroll = dlsym(ret->dlhandle, "sk_enroll")) == NULL) {
+	if (ret->dlhandle && (ret->sk_enroll = dlsym(ret->dlhandle, "sk_enroll")) == NULL) {
 		error("Security key  provider %s dlsym(sk_enroll) "
 		    "failed: %s", path, dlerror());
 		goto fail;
 	}
-	if ((ret->sk_sign = dlsym(ret->dlhandle, "sk_sign")) == NULL) {
+	if (ret->dlhandle && (ret->sk_sign = dlsym(ret->dlhandle, "sk_sign")) == NULL) {
 		error("Security key provider %s dlsym(sk_sign) failed: %s",
 		    path, dlerror());
 		goto fail;
-- 
2.23.0

