From 5dae0767e35025f6e36b5011d25813cdbb1ed866 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Thu, 17 Feb 2022 22:48:24 -0800
Subject: [PATCH] mmime: include strings.h for strncasecmp

---
 mmime.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mmime.c b/mmime.c
index f082cf0..e9a0e51 100644
--- a/mmime.c
+++ b/mmime.c
@@ -12,6 +12,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <strings.h>
 #include <time.h>
 #include <unistd.h>
 
-- 
2.34.1

