From 6ff69e64d952818f144829d87e7948bc9004ba80 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sun, 19 May 2019 13:48:45 -0700
Subject: [PATCH] bfd: Don't use long double if not available

---
 bfd/bfd.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/bfd/bfd.c b/bfd/bfd.c
index f194433883..81648b259d 100644
--- a/bfd/bfd.c
+++ b/bfd/bfd.c
@@ -862,7 +862,9 @@ union _bfd_doprnt_args
   long l;
   long long ll;
   double d;
+#if defined (__GNUC__) || defined (HAVE_LONG_DOUBLE)
   long double ld;
+#endif
   void *p;
   enum
   {
@@ -871,7 +873,9 @@ union _bfd_doprnt_args
     Long,
     LongLong,
     Double,
+#if defined (__GNUC__) || defined (HAVE_LONG_DOUBLE)
     LongDouble,
+#endif
     Ptr
   } type;
 };
@@ -1347,9 +1351,11 @@ error_handler_internal (const char *fmt, va_list ap)
 	case Double:
 	  args[i].d = va_arg (ap, double);
 	  break;
+#if defined (__GNUC__) || defined (HAVE_LONG_DOUBLE)
 	case LongDouble:
 	  args[i].ld = va_arg (ap, long double);
 	  break;
+#endif
 	case Ptr:
 	  args[i].p = va_arg (ap, void *);
 	  break;
-- 
2.30.0

