From f497d55d50cfb278a802da24e5c96682d38fa244 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sat, 21 Jan 2023 17:15:47 -0800
Subject: [PATCH] Use forward declaration instead of empty struct

---
 include/sys/zfs_context.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/sys/zfs_context.h b/include/sys/zfs_context.h
index 998eaa5dd..6d3e12002 100644
--- a/include/sys/zfs_context.h
+++ b/include/sys/zfs_context.h
@@ -382,7 +382,7 @@ typedef struct procfs_list {
 } procfs_list_t;
 
 #ifndef __cplusplus
-struct seq_file { };
+struct seq_file;
 void seq_printf(struct seq_file *m, const char *fmt, ...);
 
 typedef struct procfs_list_node {
-- 
2.44.0

