From e97b3c526b84a7bb8b504bbd765bde4924cc1bb6 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sat, 21 Jan 2023 17:52:40 -0800
Subject: [PATCH] Avoid arithmetic on void pointer

---
 module/zcommon/zfs_fletcher.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/module/zcommon/zfs_fletcher.c b/module/zcommon/zfs_fletcher.c
index 74b8c2a47..93342f615 100644
--- a/module/zcommon/zfs_fletcher.c
+++ b/module/zcommon/zfs_fletcher.c
@@ -583,7 +583,7 @@ fletcher_4_incremental_impl(boolean_t native, const void *buf, uint64_t size,
 		fletcher_4_incremental_combine(zcp, len, &nzc);
 
 		size -= len;
-		buf += len;
+		buf = (char *)buf + len;
 	}
 }
 
-- 
2.44.0

