From 0001dbe366fbdc35deb27e0144669f1912c803bd Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Tue, 5 May 2026 11:08:53 -0700
Subject: [PATCH] Use C11 alignas instead of GNU attribute

---
 include/zfs_fletcher.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/zfs_fletcher.h b/include/zfs_fletcher.h
index f3a620a7c..c45ebf99b 100644
--- a/include/zfs_fletcher.h
+++ b/include/zfs_fletcher.h
@@ -120,6 +120,7 @@ typedef void (*fletcher_4_compute_f)(fletcher_4_ctx_t *,
     const void *, uint64_t);
 
 typedef struct fletcher_4_func {
+	_Alignas(64)
 	fletcher_4_init_f init_native;
 	fletcher_4_fini_f fini_native;
 	fletcher_4_compute_f compute_native;
@@ -129,7 +130,7 @@ typedef struct fletcher_4_func {
 	boolean_t (*valid)(void);
 	boolean_t uses_fpu;
 	const char *name;
-} __attribute__((aligned(64))) fletcher_4_ops_t;
+} fletcher_4_ops_t;
 
 _ZFS_FLETCHER_H const fletcher_4_ops_t fletcher_4_superscalar_ops;
 _ZFS_FLETCHER_H const fletcher_4_ops_t fletcher_4_superscalar4_ops;
-- 
2.49.0

