From 39dc3bd751db34f04398b55e1d10a76f522b3d6b Mon Sep 17 00:00:00 2001
From: hovercats <hovercatswithlasereyes@protonmail.com>
Date: Wed, 12 Mar 2025 03:36:44 +0100
Subject: [PATCH] fix return type for main

---
 exec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/exec.c b/exec.c
index 28c5797..ce9872a 100644
--- a/exec.c
+++ b/exec.c
@@ -242,7 +242,7 @@ shuffleredir(void)
  * fabricate bootstrap code and start it (*=(argv);. -bq /usr/lib/rcmain $*)
  * start interpreting code
  */
-void
+int
 main(int argc, char *argv[])
 {
 	code bootstrap[20];
-- 
2.49.0

