From 39da3936cd5cf89e97de197181a05bb405e114b9 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Tue, 29 Apr 2025 00:48:53 -0700
Subject: [PATCH] Use complete function prototype to fix build with C23

---
 unix.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/unix.c b/unix.c
index 3f7bf9d..7052075 100644
--- a/unix.c
+++ b/unix.c
@@ -402,7 +402,7 @@ gettrap(int sig)
 	ntrap++;
 	if(ntrap>=NSIG){
 		pfmt(err, "rc: Too many traps (trap %d), dumping core\n", sig);
-		signal(SIGABRT, (void (*)())0);
+		signal(SIGABRT, (void (*)(int))0);
 		kill(getpid(), SIGABRT);
 	}
 }
@@ -411,7 +411,7 @@ void
 Trapinit(void)
 {
 	int i;
-	void (*sig)();
+	void (*sig)(int);
 
 	if(1 || flag['d']){	/* wrong!!! */
 		sig = signal(SIGINT, gettrap);
-- 
2.49.0

