From 4e822f477a9d1c71c5489e59bd30ee4b3e2f5058 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Wed, 18 May 2016 23:02:03 -0700
Subject: [PATCH] rcmain: Don't use $PLAN9/bin/9

---
 rcmain | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/rcmain b/rcmain
index 461cb3e..a73014d 100644
--- a/rcmain
+++ b/rcmain
@@ -29,11 +29,11 @@ if(flag i){
 			# is not sufficient when running in a subshell
 			# that is rc -e (like mk uses!)
 			if(builtin cd $1){
-				if(flag i) $PLAN9/bin/9 awd || status=''
+				if(flag i) awd || status=''
 				status=''
 			}
 		}
-		$PLAN9/bin/9 awd
+		awd
 	}
 	if(flag l && test -r $home/lib/profile) . $home/lib/profile
 	status=''
-- 
2.8.1

