From 0af022a4cef553bf2b09bdc6079d22adef7164ad Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sat, 10 Dec 2016 23:54:51 -0800
Subject: [PATCH] Fix sys/termios.h include to termios.h

---
 src/lib9/readcons.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lib9/readcons.c b/src/lib9/readcons.c
index 8de44b8f..289f0f3b 100644
--- a/src/lib9/readcons.c
+++ b/src/lib9/readcons.c
@@ -2,7 +2,7 @@
 #define NOPLAN9DEFINES
 #include <libc.h>
 #include <termios.h>
-#include <sys/termios.h>
+#include <termios.h>
 
 static int
 rawx(int fd, int echoing)
-- 
2.11.0

