From abdea742e220edf3b7ba8b44260e75ca95c5c9da Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sat, 10 Dec 2016 19:30:09 -0800
Subject: [PATCH] Add missing stdlib.h include for mkstemp

---
 io.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/io.c b/io.c
index 16a0709..99fa0a8 100644
--- a/io.c
+++ b/io.c
@@ -9,6 +9,7 @@
 #include <ctype.h>
 #include <errno.h>
 #include <fcntl.h>
+#include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
 
-- 
2.11.0

