From 627a4c7b6869a4ba2b15308355082f39d2263b40 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sun, 11 Dec 2016 00:14:16 -0800
Subject: [PATCH] Convert cid to char * for strlcpy

---
 sdhcp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sdhcp.c b/sdhcp.c
index 5f829ad..c5f986a 100644
--- a/sdhcp.c
+++ b/sdhcp.c
@@ -466,7 +466,7 @@ main(int argc, char *argv[])
 	if (argc)
 		ifname = argv[0]; /* interface name */
 	if (argc >= 2)
-		strlcpy(cid, argv[1], sizeof(cid)); /* client-id */
+		strlcpy((char *)cid, argv[1], sizeof(cid)); /* client-id */
 
 	memset(&ifreq, 0, sizeof(ifreq));
 	signal(SIGALRM, nop);
-- 
2.11.0

