From 9f83cd2e4af19212d26cdacf07d4b86262106383 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Tue, 2 Jul 2019 23:49:18 -0700
Subject: [PATCH] Prevent empty top-level declarations

---
 common/_elftc.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/common/_elftc.h b/common/_elftc.h
index c3df36cb..f0da3d00 100644
--- a/common/_elftc.h
+++ b/common/_elftc.h
@@ -299,7 +299,7 @@ struct name {							\
 #if defined(__GNUC__)
 #define	ELFTC_VCSID(ID)		__asm__(".ident\t\"" ID "\"")
 #else
-#define	ELFTC_VCSID(ID)		/**/
+#define	ELFTC_VCSID(ID)		_Static_assert(1, "")
 #endif
 #endif
 
@@ -307,19 +307,19 @@ struct name {							\
 #if defined(__GNUC__)
 #define	ELFTC_VCSID(ID)		__asm__(".ident\t\"" ID "\"")
 #else
-#define	ELFTC_VCSID(ID)		/**/
+#define	ELFTC_VCSID(ID)		_Static_assert(1, "")
 #endif	/* __GNU__ */
 #endif
 
 #if defined(__NetBSD__)
-#define	ELFTC_VCSID(ID)		__RCSID(ID)
+#define	ELFTC_VCSID(ID)		_Static_assert(1, "")
 #endif
 
 #if defined(__OpenBSD__)
 #if defined(__GNUC__)
 #define	ELFTC_VCSID(ID)		__asm__(".ident\t\"" ID "\"")
 #else
-#define	ELFTC_VCSID(ID)		/**/
+#define	ELFTC_VCSID(ID)		_Static_assert(1, "")
 #endif	/* __GNUC__ */
 #endif
 
-- 
2.22.0

