From 39bf781c0bf997fe434103b5a1f01765e9bc88e4 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Wed, 3 Jul 2019 00:52:16 -0700
Subject: [PATCH] Fix build without symbol versioning

---
 include/fuse.h | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/include/fuse.h b/include/fuse.h
index a92d5ee..d8e923d 100644
--- a/include/fuse.h
+++ b/include/fuse.h
@@ -919,7 +919,11 @@ void fuse_lib_help(struct fuse_args *args);
  *            `struct fuse_operations.init` handler.
  * @return the created FUSE handle
  */
-#if FUSE_USE_VERSION == 30
+#if FUSE_USE_VERSION == 31
+struct fuse *fuse_new_31(struct fuse_args *args, const struct fuse_operations *op,
+			 size_t op_size, void *private_data);
+#define fuse_new(args, op, size, data) fuse_new_31(args, op, size, data)
+#elif FUSE_USE_VERSION == 30
 struct fuse *fuse_new_30(struct fuse_args *args, const struct fuse_operations *op,
 			 size_t op_size, void *private_data);
 #define fuse_new(args, op, size, data) fuse_new_30(args, op, size, data)
-- 
2.25.1

