From 8029fd64b5b3f2c64b74fd65ec62b80bae773ae8 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Tue, 26 Oct 2021 15:27:31 -0700
Subject: [PATCH] use preferred syntactic location for epoll_event packed
 attribute

---
 include/sys/epoll.h | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/include/sys/epoll.h b/include/sys/epoll.h
index 5f975c4a..bdb43201 100644
--- a/include/sys/epoll.h
+++ b/include/sys/epoll.h
@@ -46,14 +46,14 @@ typedef union epoll_data {
 	uint64_t u64;
 } epoll_data_t;
 
-struct epoll_event {
-	uint32_t events;
-	epoll_data_t data;
-}
+struct
 #ifdef __x86_64__
 __attribute__ ((__packed__))
 #endif
-;
+epoll_event {
+	uint32_t events;
+	epoll_data_t data;
+};
 
 struct epoll_params {
 	uint32_t busy_poll_usecs;
-- 
2.49.0

