From f5847ab03a96b9b7b54fb9120f09f696a40332a1 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Fri, 18 Feb 2022 01:51:13 -0800
Subject: [PATCH] nfc: uapi: remove stray semicolon in struct declaration

Signed-off-by: Michael Forney <mforney@mforney.org>
---
 include/uapi/linux/nfc.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/uapi/linux/nfc.h b/include/uapi/linux/nfc.h
index 2f5b4be25261..bfc3a91d5a22 100644
--- a/include/uapi/linux/nfc.h
+++ b/include/uapi/linux/nfc.h
@@ -280,7 +280,7 @@ struct sockaddr_nfc_llcp {
 	__u32 nfc_protocol;
 	__u8 dsap; /* Destination SAP, if known */
 	__u8 ssap; /* Source SAP to be bound to */
-	char service_name[NFC_LLCP_MAX_SERVICE_NAME]; /* Service name URI */;
+	char service_name[NFC_LLCP_MAX_SERVICE_NAME]; /* Service name URI */
 	__kernel_size_t service_name_len;
 };
 
-- 
2.49.0

