From d9642e90c3ece95407627c005dee51d193305398 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sun, 20 Nov 2016 18:41:36 -0800
Subject: [PATCH] Handle -n option like gzip does

---
 pigz.1 | 2 +-
 pigz.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/pigz.1 b/pigz.1
index 9679e40..4e5d9f0 100644
--- a/pigz.1
+++ b/pigz.1
@@ -156,7 +156,7 @@ Display the
 license and quit.
 .TP
 .B -n --no-name
-Do not store or restore file name in/from header.
+Do not store or restore file name and mod time in/from header.
 .TP
 .B -N --name
 Store/restore file name and mod time in/from header.
diff --git a/pigz.c b/pigz.c
index a35a645..cf53623 100644
--- a/pigz.c
+++ b/pigz.c
@@ -4028,7 +4028,7 @@ local int option(char *arg)
             case 'i':  g.setdict = 0;  break;
             case 'k':  g.keep = 1;  break;
             case 'l':  g.list = 1;  break;
-            case 'n':  g.headis &= ~5;  break;
+            case 'n':  g.headis &= ~0xf;  break;
             case 'p':  get = 2;  break;
             case 'q':  g.verbosity = 0;  break;
             case 'r':  g.recurse = 1;  break;
-- 
2.10.2

