From ffccc08ae58e231589224d0297473d96e11e4916 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Tue, 26 Dec 2017 23:44:19 -0800
Subject: [PATCH] Make try_throw_ _Noreturn

---
 try.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/try.h b/try.h
index 03289dd..7d78ffe 100644
--- a/try.h
+++ b/try.h
@@ -418,7 +418,7 @@ struct try_s_ {
    to make use of any arguments after the 0 anyway.
 
    try.c must be compiled and linked to provide the try_throw_() function. */
-void try_throw_(int code, char *fmt, ...);
+_Noreturn void try_throw_(int code, char *fmt, ...);
 #define TRY_THROW_(...) try_throw_(__VA_ARGS__, NULL)
 
 /* Retry the try block.  This will start over at the beginning of the try
-- 
2.15.1

