From e8c1ae5d700e447d21db95d0be0f64fd634fa9a6 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Wed, 19 Jun 2019 20:10:09 -0700
Subject: [PATCH] Only use gcc pragmas with gcc

---
 pigz.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/pigz.c b/pigz.c
index e5a094a..a65e243 100644
--- a/pigz.c
+++ b/pigz.c
@@ -3800,8 +3800,10 @@ local char *justname(char *path) {
     return p == NULL ? path : p + 1;
 }
 
+#ifdef __GNUC__
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wunused-result"
+#endif
 
 // Copy file attributes, from -> to, as best we can. This is best effort, so no
 // errors are reported. The mode bits, including suid, sgid, and the sticky bit
@@ -3829,7 +3831,9 @@ local void copymeta(char *from, char *to) {
     (void)utimes(to, times);
 }
 
+#ifdef __GNUC__
 #pragma GCC diagnostic pop
+#endif
 
 // Set the access and modify times of fd to t.
 local void touch(char *path, time_t t) {
-- 
2.30.0

