From 08fa01f1e9cc564b9ace1872823fea40cd41d717 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sun, 5 Jun 2016 18:24:31 -0700
Subject: [PATCH] Remove remaining uses of glib

---
 sshfs.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/sshfs.c b/sshfs.c
index 472f98c..a7bcc87 100644
--- a/sshfs.c
+++ b/sshfs.c
@@ -44,7 +44,6 @@
 #include <poll.h>
 #include <netinet/in.h>
 #include <netinet/tcp.h>
-#include <glib.h>
 #include <pwd.h>
 #include <grp.h>
 #include <limits.h>
-- 
2.24.0

