From cabb791c3fb5f73b12132f8236216f508d4612bb Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sun, 5 Jun 2016 18:24:31 -0700
Subject: [PATCH] Remove remaining uses of glib

---
 sshfs.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/sshfs.c b/sshfs.c
index e027265..b035d65 100644
--- a/sshfs.c
+++ b/sshfs.c
@@ -44,7 +44,6 @@
 #include <sys/poll.h>
 #include <netinet/in.h>
 #include <netinet/tcp.h>
-#include <glib.h>
 #include <pwd.h>
 #include <grp.h>
 #include <limits.h>
-- 
2.13.2

