From 4bbbae8849274c24b3fdbd938876ef0c1afa47f7 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sun, 5 Jun 2016 18:24:31 -0700
Subject: [PATCH] Remove remaining uses of glib

---
 sshfs.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/sshfs.c b/sshfs.c
index 0b12ef6..2b274bd 100644
--- a/sshfs.c
+++ b/sshfs.c
@@ -44,7 +44,6 @@
 #include <sys/poll.h>
 #include <netinet/in.h>
 #include <netinet/tcp.h>
-#include <glib.h>
 #include <pwd.h>
 #include <grp.h>
 #include <limits.h>
@@ -4014,7 +4013,6 @@ int main(int argc, char *argv[])
 		memset(sshfs_program_path, 0, PATH_MAX);
 	}
 #endif /* __APPLE__ */
-	g_thread_init(NULL);
 
 	sshfs.blksize = 4096;
 	/* SFTP spec says all servers should allow at least 32k I/O */
-- 
2.13.1

