From a884734005e08d9e9f65d12c51a7795cab3da8a1 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sun, 9 Jul 2017 02:20:20 -0700
Subject: [PATCH] Fix poll.h include

---
 sshfs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sshfs.c b/sshfs.c
index b035d65..ab5bfd8 100644
--- a/sshfs.c
+++ b/sshfs.c
@@ -41,7 +41,7 @@
 #include <sys/utsname.h>
 #include <sys/mman.h>
 #include <sys/param.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <netinet/in.h>
 #include <netinet/tcp.h>
 #include <pwd.h>
-- 
2.13.2

