From 79aac49a3d3953b6e62b477bd488c3b7237a5aee Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Fri, 5 Jul 2019 20:52:24 -0700
Subject: [PATCH] Don't return expression in void function

---
 src/defs.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/defs.h b/src/defs.h
index 89ec13fc5..465d58139 100644
--- a/src/defs.h
+++ b/src/defs.h
@@ -824,7 +824,7 @@ extern bool pathtrace_match_set(struct tcb *, struct path_set *,
 static inline void
 pathtrace_select(const char *path)
 {
-	return pathtrace_select_set(path, &global_path_set);
+	pathtrace_select_set(path, &global_path_set);
 }
 
 static inline bool
-- 
2.44.0

