From f19be9dcb5afff4fab29d629eb9e0bcc4d6975a7 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Fri, 5 Jul 2019 20:52:24 -0700
Subject: [PATCH] Don't return expression in void function

---
 defs.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/defs.h b/defs.h
index 512ad51f..2dd5d65b 100644
--- a/defs.h
+++ b/defs.h
@@ -609,7 +609,7 @@ extern bool pathtrace_match_set(struct tcb *, struct path_set *);
 static inline void
 pathtrace_select(const char *path)
 {
-	return pathtrace_select_set(path, &global_path_set);
+	pathtrace_select_set(path, &global_path_set);
 }
 
 static inline bool
@@ -754,8 +754,8 @@ static inline void
 printxval_dispatch(const struct xlat *xlat, size_t xlat_size, uint64_t val,
 		   const char *dflt, enum xlat_type xt)
 {
-	return printxval_dispatch_ex(xlat, xlat_size, val, dflt, xt,
-				     XLAT_STYLE_DEFAULT);
+	printxval_dispatch_ex(xlat, xlat_size, val, dflt, xt,
+			      XLAT_STYLE_DEFAULT);
 }
 
 enum xlat_style_private_flag_bits {
-- 
2.22.0

