From 16cb24454fc4a2aa3971507c96a7c985fe39a041 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Fri, 5 Jul 2019 20:52:24 -0700
Subject: [PATCH] Don't return expression in void function

---
 defs.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/defs.h b/defs.h
index 0968bc35..961b84fe 100644
--- a/defs.h
+++ b/defs.h
@@ -676,7 +676,7 @@ extern bool pathtrace_match_set(struct tcb *, struct path_set *);
 static inline void
 pathtrace_select(const char *path)
 {
-	return pathtrace_select_set(path, &global_path_set);
+	pathtrace_select_set(path, &global_path_set);
 }
 
 static inline bool
-- 
2.26.0

