From 5c0b5a421c4dd366e75945fbcba2591835d6e7ed Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Fri, 5 Jul 2019 20:52:24 -0700
Subject: [PATCH] Don't return expression in void function

---
 defs.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/defs.h b/defs.h
index 7151e90c..656c04a5 100644
--- a/defs.h
+++ b/defs.h
@@ -704,7 +704,7 @@ extern bool pathtrace_match_set(struct tcb *, struct path_set *);
 static inline void
 pathtrace_select(const char *path)
 {
-	return pathtrace_select_set(path, &global_path_set);
+	pathtrace_select_set(path, &global_path_set);
 }
 
 static inline bool
-- 
2.28.0

