From dc7f55485c8562999b5bc2ae713a3aec0c90bce8 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sun, 7 Jul 2019 21:53:16 -0700
Subject: [PATCH] Don't omit second operand to '?' operator

---
 open.c   | 5 +++--
 prctl.c  | 2 +-
 strace.c | 6 +++---
 3 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/open.c b/open.c
index 690453ed..1960ea1c 100644
--- a/open.c
+++ b/open.c
@@ -77,8 +77,9 @@ sprint_open_modes64(uint64_t flags)
 	}
 	*p = '\0';
 
-	return sprintflags_ex(outstr, open_mode_flags, flags, sep,
-			      XLAT_STYLE_ABBREV) ?: outstr;
+	str = sprintflags_ex(outstr, open_mode_flags, flags, sep,
+			     XLAT_STYLE_ABBREV);
+	return str ? str : outstr;
 }
 
 const char *
diff --git a/prctl.c b/prctl.c
index b904fa34..1366a89e 100644
--- a/prctl.c
+++ b/prctl.c
@@ -76,7 +76,7 @@ sprint_sve_val(kernel_ulong_t arg)
 	const char *flags_str = sprintflags("", pr_sve_vl_flags, flags);
 
 	xsprintf(out, "%s%s%#" PRI_klx,
-		 flags_str ?: "", flags_str ? "|" : "", vl);
+		 flags_str ? flags_str : "", flags_str ? "|" : "", vl);
 
 	return out;
 }
diff --git a/strace.c b/strace.c
index 2a4e6001..352c8282 100644
--- a/strace.c
+++ b/strace.c
@@ -2235,7 +2235,7 @@ init(int argc, char *argv[])
 			break;
 		case GETOPT_TS:
 			tflag_long_set = true;
-			if (parse_ts_arg(optarg ?: tflag_str))
+			if (parse_ts_arg(optarg ? optarg : tflag_str))
 				error_opt_arg(c, lopt, optarg);
 			break;
 		case 'T':
@@ -2332,10 +2332,10 @@ init(int argc, char *argv[])
 			qualify_kvm(optarg);
 			break;
 		case GETOPT_QUAL_QUIET:
-			qualify_quiet(optarg ?: qflag_qual);
+			qualify_quiet(optarg ? optarg : qflag_qual);
 			break;
 		case GETOPT_QUAL_DECODE_FD:
-			qualify_decode_fd(optarg ?: yflag_qual);
+			qualify_decode_fd(optarg ? optarg : yflag_qual);
 			break;
 		default:
 			error_msg_and_help(NULL);
-- 
2.26.2

