From 0ad7047e306f4aef8f4a6346e9dfff8079b1c571 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Thu, 30 Jan 2020 21:57:34 -0800
Subject: [PATCH] Use alloca when VLAs aren't available

---
 evdev.c | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/evdev.c b/evdev.c
index 1d4e9aef..58e705ae 100644
--- a/evdev.c
+++ b/evdev.c
@@ -181,9 +181,13 @@ decode_bitset(struct tcb *const tcp, const kernel_ulong_t arg,
 		return RVAL_IOCTL_DECODED;
 	}
 
+#ifndef __STDC_NO_VLA__
 	char decoded_arg[size];
+#else
+	char *decoded_arg = alloca(size);
+#endif
 
-	if (umove_or_printaddr(tcp, arg, &decoded_arg))
+	if (umoven_or_printaddr(tcp, arg, size, decoded_arg))
 		return RVAL_IOCTL_DECODED;
 
 	if (xlat_verbose(xlat_verbosity) != XLAT_STYLE_RAW) {
@@ -238,9 +242,13 @@ mtslots_ioctl(struct tcb *const tcp, const unsigned int code,
 		return RVAL_IOCTL_DECODED;
 	}
 
+#ifndef __STDC_NO_VLA__
 	int buffer[size];
+#else
+	int *buffer = alloca(size * sizeof(int));
+#endif
 
-	if (umove_or_printaddr(tcp, arg, &buffer))
+	if (umoven_or_printaddr(tcp, arg, size * sizeof(int), &buffer))
 		return RVAL_IOCTL_DECODED;
 
 	tprints("{code=");
@@ -249,7 +257,7 @@ mtslots_ioctl(struct tcb *const tcp, const unsigned int code,
 	tprints(", values=[");
 
 	unsigned int i;
-	for (i = 1; i < ARRAY_SIZE(buffer); i++)
+	for (i = 1; i < size; i++)
 		tprintf("%s%d", i > 1 ? ", " : "", buffer[i]);
 
 	tprints("]}");
-- 
2.28.0

