From c30b64a5e7bf9f196a60727a5886a78ecd53d9f5 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Mon, 28 Jan 2019 17:00:54 -0800
Subject: [PATCH] Define _GNU_SOURCE for exp10

---
 alsamixer/volume_mapping.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/alsamixer/volume_mapping.c b/alsamixer/volume_mapping.c
index 94bd0fe..3e65e0f 100644
--- a/alsamixer/volume_mapping.c
+++ b/alsamixer/volume_mapping.c
@@ -31,6 +31,7 @@
  */
 
 #define _ISOC99_SOURCE /* lrint() */
+#define _GNU_SOURCE /* exp10 */
 #include "aconfig.h"
 #include <math.h>
 #include <stdbool.h>
-- 
2.20.1

