From a13b37473388ed731c2dc96c6bf2b053a4de26c3 Mon Sep 17 00:00:00 2001
From: Mark Nudelman <markn@greenwoodsoftware.com>
Date: Tue, 27 Jul 2021 10:33:14 -0700
Subject: [PATCH] Fix compiler warning.

---
 funcs.h | 2 +-
 xbuf.c  | 2 +-
 xbuf.h  | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/funcs.h b/funcs.h
index 330540a..29de5d3 100644
--- a/funcs.h
+++ b/funcs.h
@@ -351,4 +351,4 @@ public int getchr LESSPARAMS ((VOID_PARAM));
 public void xbuf_init LESSPARAMS ((struct xbuffer *xbuf));
 public void xbuf_deinit LESSPARAMS ((struct xbuffer *xbuf));
 public void xbuf_reset LESSPARAMS ((struct xbuffer *xbuf));
-public void xbuf_add LESSPARAMS ((struct xbuffer *xbuf, char ch));
+public void xbuf_add LESSPARAMS ((struct xbuffer *xbuf, int ch));
diff --git a/xbuf.c b/xbuf.c
index f20dfae..cc6ec47 100644
--- a/xbuf.c
+++ b/xbuf.c
@@ -34,7 +34,7 @@ xbuf_reset(xbuf)
 	public void
 xbuf_add(xbuf, ch)
 	struct xbuffer *xbuf;
-	char ch;
+	int ch;
 {
 	if (xbuf->end >= xbuf->size)
 	{
diff --git a/xbuf.h b/xbuf.h
index ba62ef1..b6bd160 100644
--- a/xbuf.h
+++ b/xbuf.h
@@ -10,6 +10,6 @@ struct xbuffer
 
 void xbuf_init(struct xbuffer *xbuf);
 void xbuf_reset(struct xbuffer *xbuf);
-void xbuf_add(struct xbuffer *xbuf, char ch);
+void xbuf_add(struct xbuffer *xbuf, int ch);
 
 #endif
-- 
2.32.0

