From 380ddf6615e166c0d8b0ba3889783947c39d5bef Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Wed, 3 Jan 2018 19:22:18 -0800
Subject: [PATCH] Remove sys/cdefs.h include and usage

---
 lib/fixedsizehash.h | 1 -
 libelf/elf.h        | 4 ----
 2 files changed, 5 deletions(-)

diff --git a/lib/fixedsizehash.h b/lib/fixedsizehash.h
index dac2a5f5..43016fc3 100644
--- a/lib/fixedsizehash.h
+++ b/lib/fixedsizehash.h
@@ -30,7 +30,6 @@
 #include <errno.h>
 #include <stdlib.h>
 #include <string.h>
-#include <sys/cdefs.h>
 
 #include <system.h>
 
diff --git a/libelf/elf.h b/libelf/elf.h
index 75043bcb..23b9a789 100644
--- a/libelf/elf.h
+++ b/libelf/elf.h
@@ -21,8 +21,6 @@
 
 #include <features.h>
 
-__BEGIN_DECLS
-
 /* Standard ELF types.  */
 
 #include <stdint.h>
@@ -3915,6 +3913,4 @@ enum
 #define R_METAG_TLS_LE_HI16	60
 #define R_METAG_TLS_LE_LO16	61
 
-__END_DECLS
-
 #endif	/* elf.h */
-- 
2.18.0

