From 196586b33706fc3d8efaf51d60676be307f525e3 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sat, 30 May 2020 00:24:00 -0700
Subject: [PATCH] Remove unnecessary alloca

---
 lib/common/io_stdin.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/lib/common/io_stdin.c b/lib/common/io_stdin.c
index 4694989..6451f6b 100644
--- a/lib/common/io_stdin.c
+++ b/lib/common/io_stdin.c
@@ -43,7 +43,7 @@ static int stdin_read_at(sqfs_file_t *base, sqfs_u64 offset,
 {
 	sqfs_file_stdinout_t *file = (sqfs_file_stdinout_t *)base;
 	size_t temp_size = 0;
-	sqfs_u8 *temp = NULL;
+	sqfs_u8 *temp = NULL, temp_buf[1024];
 	sqfs_u64 diff;
 	size_t ret;
 
@@ -51,8 +51,8 @@ static int stdin_read_at(sqfs_file_t *base, sqfs_u64 offset,
 		return SQFS_ERROR_IO;
 
 	if (offset > file->offset) {
-		temp_size = 1024;
-		temp = alloca(temp_size);
+		temp_size = sizeof(temp_buf);
+		temp = temp_buf;
 	}
 
 	if (offset >= file->real_size || (offset + size) > file->real_size)
-- 
2.26.2

