From d7dd64b56ce96630d01a269439b86dc940c9df07 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sat, 30 May 2020 00:24:10 -0700
Subject: [PATCH] Add missing includes of alloca.h

---
 bin/sqfsdiff/extract.c | 2 ++
 lib/common/mkdir_p.c   | 1 +
 2 files changed, 3 insertions(+)

diff --git a/bin/sqfsdiff/extract.c b/bin/sqfsdiff/extract.c
index 979572a..45e32e7 100644
--- a/bin/sqfsdiff/extract.c
+++ b/bin/sqfsdiff/extract.c
@@ -6,6 +6,8 @@
  */
 #include "sqfsdiff.h"
 
+#include <alloca.h>
+
 static int extract(sqfs_data_reader_t *data, const sqfs_inode_generic_t *inode,
 		   const char *prefix, const char *path, size_t block_size)
 {
diff --git a/lib/common/mkdir_p.c b/lib/common/mkdir_p.c
index 0413495..b1845b1 100644
--- a/lib/common/mkdir_p.c
+++ b/lib/common/mkdir_p.c
@@ -6,6 +6,7 @@
  */
 #include "common.h"
 
+#include <alloca.h>
 #include <string.h>
 #include <stdlib.h>
 #include <stdio.h>
-- 
2.26.2

