From 1bcd4df6fbcd9c3dbace99b62065b05a75653d4b Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sat, 30 May 2020 01:45:49 -0700
Subject: [PATCH] Add mul/add overflow fallbacks

---
 include/compat.h | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/include/compat.h b/include/compat.h
index 3eac307..5a879e2 100644
--- a/include/compat.h
+++ b/include/compat.h
@@ -30,7 +30,17 @@
 #		define SZ_MUL_OV __builtin_mul_overflow
 #	endif
 #else
-#	error I do not know how to trap integer overflows with this compiler
+static inline bool SZ_ADD_OV(size_t a, size_t b, size_t *res)
+{
+	*res = a + b;
+	return a > SIZE_MAX - b;
+}
+
+static inline bool SZ_MUL_OV(size_t a, size_t b, size_t *res)
+{
+	*res = a * b;
+	return b && a > SIZE_MAX / b;
+}
 #endif
 
 #if defined(_WIN32) || defined(__WINDOWS__)
-- 
2.26.2

