From e25b260070a79b63f8f6be0cfcfcd1bd48f5e75c Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Thu, 6 Jul 2017 20:32:01 -0700
Subject: [PATCH] Remove __{BEGIN,END}_DECLS and add sys/types include

---
 extern.h | 6 ++----
 http.h   | 4 ----
 2 files changed, 2 insertions(+), 8 deletions(-)

diff --git a/extern.h b/extern.h
index cb7664e..3d43c05 100644
--- a/extern.h
+++ b/extern.h
@@ -17,6 +17,8 @@
 #ifndef EXTERN_H
 #define EXTERN_H
 
+#include <sys/types.h>
+
 #ifndef PATH_VAR_EMPTY
 #define	PATH_VAR_EMPTY "/var/empty"
 #endif
@@ -173,8 +175,6 @@ struct	capaths {
 
 struct	jsmnn;
 
-__BEGIN_DECLS
-
 /*
  * Start with our components.
  * These are all isolated and talk to each other using sockets.
@@ -267,6 +267,4 @@ int		 verbose;
  */
 enum comp	 proccomp;
 
-__END_DECLS
-
 #endif /* ! EXTERN_H */
diff --git a/http.h b/http.h
index ba88107..af196db 100644
--- a/http.h
+++ b/http.h
@@ -62,8 +62,6 @@ struct	httpget {
 	size_t		 bodypartsz; /* size of bodypart */
 };
 
-__BEGIN_DECLS
-
 /* Configuration functions. */
 struct httpcfg	*http_init(void);
 void		 http_uninit(struct httpcfg *);
@@ -96,6 +94,4 @@ int		 http_head_status(const struct http *,
 struct httphead	*http_head_get(const char *,
 			struct httphead *, size_t);
 
-__END_DECLS
-
 #endif /* HTTP_H */
-- 
2.13.2

