From ba5389200ff60bb5494efe503f1f1fc165383a85 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Fri, 23 Dec 2016 14:41:52 -0800
Subject: [PATCH] Remove non-reproducible info from about page

---
 content/fetchers/about.c | 17 -----------------
 1 file changed, 17 deletions(-)

diff --git a/content/fetchers/about.c b/content/fetchers/about.c
index 4d14020b0..aff21c81a 100644
--- a/content/fetchers/about.c
+++ b/content/fetchers/about.c
@@ -477,23 +477,6 @@ static bool fetch_about_testament_handler(struct fetch_about_context *ctx)
 	if (fetch_about_send_callback(&msg, ctx))
 		goto fetch_about_testament_handler_aborted;
 
-
-	slen = snprintf(buffer, sizeof buffer,
-			"Built by %s (%s) from %s at revision %s on %s\n\n",
-			GECOS, USERNAME, WT_BRANCHPATH, WT_REVID, WT_COMPILEDATE);
-
-	msg.data.header_or_data.len = slen;
-	if (fetch_about_send_callback(&msg, ctx))
-		goto fetch_about_testament_handler_aborted;
-
-	slen = snprintf(buffer, sizeof buffer,
-			"Built on %s in %s\n\n",
-			WT_HOSTNAME, WT_ROOT);
-
-	msg.data.header_or_data.len = slen;
-	if (fetch_about_send_callback(&msg, ctx))
-		goto fetch_about_testament_handler_aborted;
-
 	if (WT_MODIFIED > 0) {
 		slen = snprintf(buffer, sizeof buffer,
 				"Working tree has %d modification%s\n\n",
-- 
2.11.0

