From d83c405659c8516cb32a1e57dcfbb1752d26319d Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sun, 28 May 2017 16:34:32 -0700
Subject: [PATCH] Add missing sys/xattr.h include

---
 misc/create_inode.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/misc/create_inode.c b/misc/create_inode.c
index 8ce3fafa..a972f936 100644
--- a/misc/create_inode.c
+++ b/misc/create_inode.c
@@ -20,6 +20,8 @@
 #include <limits.h> /* for PATH_MAX */
 #ifdef HAVE_ATTR_XATTR_H
 #include <attr/xattr.h>
+#else
+#include <sys/xattr.h>
 #endif
 #include <sys/ioctl.h>
 #ifdef HAVE_SYS_SYSMACROS_H
-- 
2.14.1

