From 568a13cb63d8d6a33c7ef77ec9c75e3abded1be3 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Wed, 19 Jun 2019 21:34:47 -0700
Subject: [PATCH] Remove _ALIGNED after struct definition

---
 include/dom/core/string.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/dom/core/string.h b/include/dom/core/string.h
index f2a6122..4c2da97 100644
--- a/include/dom/core/string.h
+++ b/include/dom/core/string.h
@@ -18,7 +18,7 @@
 typedef struct dom_string dom_string;
 struct dom_string {
 	uint32_t refcnt;
-} _ALIGNED;
+};
 
 
 /* Claim a reference on a DOM string */
-- 
2.20.1

