From 4919efdefac11325db27c6a848e381a7cc28eacd Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Thu, 2 Apr 2026 17:36:09 -0700
Subject: [PATCH] Use prototype for getenv declaration

---
 lib/fnmatch.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/fnmatch.c b/lib/fnmatch.c
index 01da376..cb1c856 100644
--- a/lib/fnmatch.c
+++ b/lib/fnmatch.c
@@ -121,7 +121,7 @@ USA.  */
    whose names are inconsistent.  */
 
 # if !defined _LIBC && !defined getenv
-extern char *getenv ();
+extern char *getenv (const char *);
 # endif
 
 # ifndef errno
-- 
2.49.0

