From d0cf1c91f842993121bc24874153f5959dbd95f3 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Tue, 7 Sep 2021 01:43:25 -0700
Subject: [PATCH] libavformat: Use C99 __func__ instead of non-standard
 __FUNCTION__

---
 libavformat/mux.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavformat/mux.c b/libavformat/mux.c
index 31361f9b46..8dcf4047dc 100644
--- a/libavformat/mux.c
+++ b/libavformat/mux.c
@@ -1108,7 +1108,7 @@ static int write_packet_common(AVFormatContext *s, AVStream *st, AVPacket *pkt,
     int ret;
 
     if (s->debug & FF_FDEBUG_TS)
-        av_log(s, AV_LOG_DEBUG, "%s size:%d dts:%s pts:%s\n", __FUNCTION__,
+        av_log(s, AV_LOG_DEBUG, "%s size:%d dts:%s pts:%s\n", __func__,
                pkt->size, av_ts2str(pkt->dts), av_ts2str(pkt->pts));
 
     guess_pkt_duration(s, st, pkt);
-- 
2.37.3

