From be0fa40e9d56ff85c3329ce7452bf33215fbd678 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Mon, 25 Oct 2021 17:08:29 -0700
Subject: [PATCH] libavutil: Use C11 _Alignas as fallback for DECLARE_ALIGNED

---
 libavutil/mem.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libavutil/mem.h b/libavutil/mem.h
index d91174196c..996e9b14b0 100644
--- a/libavutil/mem.h
+++ b/libavutil/mem.h
@@ -120,6 +120,10 @@
     #define DECLARE_ALIGNED(n,t,v)      __declspec(align(n)) t v
     #define DECLARE_ASM_ALIGNED(n,t,v)  __declspec(align(n)) t v
     #define DECLARE_ASM_CONST(n,t,v)    __declspec(align(n)) static const t v
+#elif defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L
+    #define DECLARE_ALIGNED(n,t,v)      _Alignas(n) t v
+    #define DECLARE_ASM_ALIGNED(n,t,v)  _Alignas(n) t v
+    #define DECLARE_ASM_CONST(n,t,v)    static const _Alignas(n) t v
 #else
     #define DECLARE_ALIGNED(n,t,v)      t v
     #define DECLARE_ASM_ALIGNED(n,t,v)  t v
-- 
2.37.3

