From 788b21499ade6f5f03a4c79f7a8a82e9af873af4 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Mon, 25 Oct 2021 17:08:29 -0700
Subject: [PATCH] libavutil: Use C11 _Alignas as fallback for DECLARE_ALIGNED

---
 libavutil/mem_internal.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libavutil/mem_internal.h b/libavutil/mem_internal.h
index 2448c606f1..dd037fd3b5 100644
--- a/libavutil/mem_internal.h
+++ b/libavutil/mem_internal.h
@@ -91,6 +91,10 @@
     #define DECLARE_ALIGNED(n,t,v)      __declspec(align(n)) t v
     #define DECLARE_ASM_ALIGNED(n,t,v)  __declspec(align(n)) t v
     #define DECLARE_ASM_CONST(n,t,v)    __declspec(align(n)) static const t v
+#elif defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L
+    #define DECLARE_ALIGNED(n,t,v)      _Alignas(n) t v
+    #define DECLARE_ASM_ALIGNED(n,t,v)  _Alignas(n) t v
+    #define DECLARE_ASM_CONST(n,t,v)    static const _Alignas(n) t v
 #else
     #define DECLARE_ALIGNED(n,t,v)      t v
     #define DECLARE_ASM_ALIGNED(n,t,v)  t v
-- 
2.37.3

