From befefcc3c3fef5d7d15713ae9dfb82108c331d56 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sun, 11 Aug 2019 04:50:32 +0000
Subject: [PATCH] Don't return expression in function returning void

---
 lib/cache.c  | 2 +-
 lib/object.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/cache.c b/lib/cache.c
index 427f5df..badf91c 100644
--- a/lib/cache.c
+++ b/lib/cache.c
@@ -421,7 +421,7 @@ void nl_cache_free(struct nl_cache *cache)
 
 void nl_cache_put(struct nl_cache *cache)
 {
-	return nl_cache_free(cache);
+	nl_cache_free(cache);
 }
 
 /** @} */
diff --git a/lib/object.c b/lib/object.c
index aeb4307..f0c73da 100644
--- a/lib/object.c
+++ b/lib/object.c
@@ -300,7 +300,7 @@ void nl_object_dump_buf(struct nl_object *obj, char *buf, size_t len)
 		.dp_buflen = len,
 	};
 
-	return nl_object_dump(obj, &dp);
+	nl_object_dump(obj, &dp);
 }
 
 /**
-- 
2.22.0

