From 672da355b5beb5bc2729f968318f87eb7795572b Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Mon, 12 Aug 2019 02:22:15 +0000
Subject: [PATCH] Fallback to alloca when VLAs aren't available

---
 lib/genl/mngt.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lib/genl/mngt.c b/lib/genl/mngt.c
index ebfe85e..dda661f 100644
--- a/lib/genl/mngt.c
+++ b/lib/genl/mngt.c
@@ -61,7 +61,11 @@ static int cmd_msg_parser(struct sockaddr_nl *who, struct nlmsghdr *nlh,
 	if (cmd->c_msg_parser == NULL)
 		err = -NLE_OPNOTSUPP;
 	else {
+#ifndef __STDC_NO_VLA__
 		struct nlattr *tb[cmd->c_maxattr + 1];
+#else
+		struct nlattr **tb = alloca((cmd->c_maxattr + 1) * sizeof(tb[0]));
+#endif
 		struct genl_info info = {
 			.who = who,
 			.nlh = nlh,
-- 
2.22.0

