From 2e94ef2986ff25f4b9d9e5d14fd3b7e2f889eed0 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sun, 2 Feb 2025 03:01:17 -0800
Subject: [PATCH] Set LUA_ROOT to "/"

---
 luaconf.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/luaconf.h b/src/luaconf.h
index 96a77802..b82e483d 100644
--- a/src/luaconf.h
+++ b/src/luaconf.h
@@ -239,7 +239,7 @@
 
 #else			/* }{ */
 
-#define LUA_ROOT	"/usr/local/"
+#define LUA_ROOT	"/"
 #define LUA_LDIR	LUA_ROOT "share/lua/" LUA_VDIR "/"
 #define LUA_CDIR	LUA_ROOT "lib/lua/" LUA_VDIR "/"
 
-- 
2.49.0

