From ac5d9c9d271023f1d0a9733946f7f079b7829625 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Fri, 18 Nov 2016 21:35:52 -0800
Subject: [PATCH] Allow building lpeg statically

---
 Makefile  | 3 ++-
 vis-lua.c | 9 +++++++++
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index e221b67..b28590c 100644
--- a/Makefile
+++ b/Makefile
@@ -14,6 +14,7 @@ MANPREFIX ?= ${PREFIX}/man
 VERSION = $(shell git describe --always 2>/dev/null || echo "0.2")
 
 CONFIG_LUA ?= 1
+CONFIG_LUA_STATIC ?= 0
 CONFIG_ACL ?= 0
 CONFIG_SELINUX ?= 0
 
@@ -25,7 +26,7 @@ CFLAGS_VIS = $(CFLAGS_AUTO) $(CFLAGS_TERMKEY) $(CFLAGS_CURSES) $(CFLAGS_ACL) \
 	$(CFLAGS_SELINUX) $(CFLAGS_LUA) $(CFLAGS_STD)
 
 CFLAGS_VIS += -DVIS_PATH=\"${SHAREPREFIX}/vis\"
-CFLAGS_VIS += -DCONFIG_LUA=${CONFIG_LUA}
+CFLAGS_VIS += -DCONFIG_LUA=${CONFIG_LUA} -DCONFIG_LUA_STATIC=${CONFIG_LUA_STATIC}
 CFLAGS_VIS += -DCONFIG_SELINUX=${CONFIG_SELINUX}
 CFLAGS_VIS += -DCONFIG_ACL=${CONFIG_ACL}
 CFLAGS_VIS += ${CFLAGS_DEBUG}
diff --git a/vis-lua.c b/vis-lua.c
index 5b039ba..f6ed68e 100644
--- a/vis-lua.c
+++ b/vis-lua.c
@@ -1404,6 +1404,15 @@ void vis_lua_init(Vis *vis) {
 	vis->lua = L;
 	luaL_openlibs(L);
 
+#if CONFIG_LUA_STATIC
+	extern int luaopen_lpeg(lua_State *L);
+	lua_getglobal(L, "package");
+	lua_getfield(L, -1, "preload");
+	lua_pushcfunction(L, luaopen_lpeg);
+	lua_setfield(L, -2, "lpeg");
+	lua_pop(L, 2);
+#endif
+
 	/* remove any relative paths from lua's default package.path */
 	vis_lua_path_strip(vis);
 
-- 
2.10.2

