From be040c6797873884126b25c78b72922ad9e13f4d Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Fri, 18 Nov 2016 21:53:25 -0800
Subject: [PATCH] Set default theme to dark-16

---
 vis.lua | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vis.lua b/vis.lua
index 0d14f8e..6c0e2a7 100644
--- a/vis.lua
+++ b/vis.lua
@@ -417,4 +417,4 @@ vis.events.win_status = function(win)
 	win:status(left, right);
 end
 
-vis.events.theme_change((not vis.ui or vis.ui.colors <= 16) and "default-16" or "default-256")
+vis.events.theme_change("dark-16")
-- 
2.10.2

