From 02d453299c908983ec4333e9a924023e639df19f Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Tue, 2 Jul 2019 16:04:41 -0700
Subject: [PATCH] Use #define for constants that can't be represented as int

---
 include/sound/asound.h | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/include/sound/asound.h b/include/sound/asound.h
index 4d35c84d..a616f95c 100644
--- a/include/sound/asound.h
+++ b/include/sound/asound.h
@@ -312,11 +312,9 @@ typedef int __bitwise snd_pcm_state_t;
 #define	SNDRV_PCM_STATE_DISCONNECTED	((__force snd_pcm_state_t) 8) /* hardware is disconnected */
 #define	SNDRV_PCM_STATE_LAST		SNDRV_PCM_STATE_DISCONNECTED
 
-enum {
-	SNDRV_PCM_MMAP_OFFSET_DATA = 0x00000000,
-	SNDRV_PCM_MMAP_OFFSET_STATUS = 0x80000000,
-	SNDRV_PCM_MMAP_OFFSET_CONTROL = 0x81000000,
-};
+#define SNDRV_PCM_MMAP_OFFSET_DATA	0x00000000
+#define SNDRV_PCM_MMAP_OFFSET_STATUS	0x80000000
+#define SNDRV_PCM_MMAP_OFFSET_CONTROL	0x81000000
 
 union snd_pcm_sync_id {
 	unsigned char id[16];
-- 
2.22.0

