From f0732cf0a75f47eddc200bb6fa9c91250e257a3c Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Wed, 3 Jul 2019 18:27:09 -0700
Subject: [PATCH] Don't use inline asm on non-GNU compilers

---
 include/local.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/local.h b/include/local.h
index ce142d04..df3e6c31 100644
--- a/include/local.h
+++ b/include/local.h
@@ -265,7 +265,7 @@ extern snd_lib_error_handler_t snd_err_msg;
 
 #define ASM_NAME(name) __SYMBOL_PREFIX name
 
-#ifdef HAVE_GNU_LD
+#if defined(HAVE_GNU_LD) && defined(__GNUC__)
 # ifdef HAVE_ELF
 
 /* We want the .gnu.warning.SYMBOL section to be unallocated.  */
-- 
2.22.0

