From b2c26e9b2ef3899b640fa5a70124b0a0ce42f011 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Fri, 31 May 2019 17:48:37 -0700
Subject: [PATCH] Remove unused empty struct

Signed-off-by: Michael Forney <mforney@mforney.org>
---
 src/pcm/pcm_direct.h | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/pcm/pcm_direct.h b/src/pcm/pcm_direct.h
index da5e280e..6e66b47d 100644
--- a/src/pcm/pcm_direct.h
+++ b/src/pcm/pcm_direct.h
@@ -181,8 +181,6 @@ struct snd_pcm_direct {
 			mix_areas_24_t *remix_areas_24;
 			mix_areas_u8_t *remix_areas_u8;
 		} dmix;
-		struct {
-		} dsnoop;
 		struct {
 			unsigned long long chn_mask;
 		} dshare;
-- 
2.20.1

