From 262d679517a8867f38e15369414a6cabe897ccb1 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Wed, 17 Oct 2018 16:48:16 -0700
Subject: [PATCH] Include sys/select.h from virtio.h for fd_set

---
 virtio.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/virtio.h b/virtio.h
index 92b1070..d53c8c4 100644
--- a/virtio.h
+++ b/virtio.h
@@ -24,6 +24,8 @@
 #ifndef VIRTIO_H
 #define VIRTIO_H
 
+#include <sys/select.h>
+
 #include "iomem.h"
 #include "pci.h"
 
-- 
2.19.0

