From bf6cb0cd861dd2cbfa6a2e4754b6dd278b89b1d9 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Fri, 23 Dec 2016 15:24:36 -0800
Subject: [PATCH] Use relative include paths

---
 src/duk-libdom-common.c | 14 +++-----------
 1 file changed, 3 insertions(+), 11 deletions(-)

diff --git a/src/duk-libdom-common.c b/src/duk-libdom-common.c
index 6111070..ff225b0 100644
--- a/src/duk-libdom-common.c
+++ b/src/duk-libdom-common.c
@@ -85,17 +85,9 @@ int output_tool_prologue(FILE* outf)
 {
         char *fpath;
 
-        fpath = genb_fpath("binding.h");
-        fprintf(outf, "\n#include \"%s\"\n", fpath);
-        free(fpath);
-
-        fpath = genb_fpath("private.h");
-        fprintf(outf, "#include \"%s\"\n", fpath);
-        free(fpath);
-
-        fpath = genb_fpath("prototype.h");
-        fprintf(outf, "#include \"%s\"\n", fpath);
-        free(fpath);
+        fputs("\n#include \"binding.h\"\n", outf);
+        fputs("#include \"private.h\"\n", outf);
+        fputs("#include \"prototype.h\"\n", outf);
 
         return 0;
 }
-- 
2.11.0

