From 16228e926b5718cb0012371e8351156ef1da75c9 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sat, 10 Dec 2016 19:57:39 -0800
Subject: [PATCH] maketab: Only consider #define lines that match the expected
 form

---
 maketab.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/maketab.c b/maketab.c
index c3ce5c6..9981422 100644
--- a/maketab.c
+++ b/maketab.c
@@ -138,7 +138,7 @@ int main(int argc, char *argv[])
 	while (fgets(buf, sizeof buf, fp) != NULL) {
 		// 199 is sizeof(def) - 1
 		n = sscanf(buf, "%1c %199s %199s %d", &c, def, name, &tok);
-		if (c != '#' || (n != 4 && strcmp(def,"define") != 0))	/* not a valid #define */
+		if (c != '#' || n != 4 || strcmp(def,"define") != 0)	/* not a valid #define */
 			continue;
 		if (strcmp(name, "YYSTYPE_IS_DECLARED") == 0)
 			continue;
-- 
2.24.0

