From 67cc439f1b0db14d106ea1b6b96d66c6b6c7e15a Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Mon, 18 Apr 2016 00:13:51 -0700
Subject: [PATCH] pax: Set listf to stderr in main

---
 bin/pax/pax.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/bin/pax/pax.c b/bin/pax/pax.c
index 6d70b7a..51a36c9 100644
--- a/bin/pax/pax.c
+++ b/bin/pax/pax.c
@@ -91,7 +91,7 @@ int	docrc;			/* check/create file crc */
 char	*dirptr;		/* destination dir in a copy */
 char	*argv0;			/* root of argv[0] */
 sigset_t s_mask;		/* signal mask for cleanup critical sect */
-FILE	*listf = stderr;	/* file pointer to print file list to */
+FILE	*listf;			/* file pointer to print file list to */
 int	listfd = STDERR_FILENO;	/* fd matching listf, for sighandler output */
 char	*tempfile;		/* tempfile to use for mkstemp(3) */
 char	*tempbase;		/* basename of tempfile to use for mkstemp(3) */
@@ -222,6 +222,8 @@ main(int argc, char **argv)
 	char *tmpdir;
 	size_t tdlen;
 
+	listf = stderr;
+
 	/*
 	 * Keep a reference to cwd, so we can always come back home.
 	 */
-- 
2.8.0

