From 0578803f33e11349628ff98a2834bb0dae6fc4f4 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Mon, 18 Apr 2016 01:16:12 -0700
Subject: [PATCH] tar: Default to stdin

---
 bin/pax/options.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/bin/pax/options.c b/bin/pax/options.c
index adb7f4a..89d29bb 100644
--- a/bin/pax/options.c
+++ b/bin/pax/options.c
@@ -877,9 +877,7 @@ tar_options(int argc, char **argv)
 
 	if (!fstdin && ((arcname == NULL) || (*arcname == '\0'))) {
 		arcname = getenv("TAPE");
-		if ((arcname == NULL) || (*arcname == '\0'))
-			arcname = _PATH_DEFTAPE;
-		else if ((arcname[0] == '-') && (arcname[1]== '\0')) {
+		if ((arcname == NULL) || (*arcname == '\0') || (arcname[0] == '-') && (arcname[1]== '\0')) {
 			arcname = NULL;
 			fstdin = 1;
 		}
-- 
2.8.1

