From 29891dd5402ef6d9d3a4e10e69fe9d2e953cd34b Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Fri, 18 Nov 2016 21:53:25 -0800
Subject: [PATCH] Set default theme to dark-16

---
 lua/visrc.lua | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lua/visrc.lua b/lua/visrc.lua
index 13bd7f7..e308fc3 100644
--- a/lua/visrc.lua
+++ b/lua/visrc.lua
@@ -5,7 +5,7 @@ require('plugins/textobject-lexer')
 
 vis.events.subscribe(vis.events.INIT, function()
 	-- Your global configuration options
-	vis:command("set theme ".. (vis.ui.colors <= 16 and "default-16" or "default-256"))
+	vis:command("set theme default-16")
 end)
 
 vis.events.subscribe(vis.events.WIN_OPEN, function(win)
-- 
2.11.0

