From c1dc134f5f2591d364907a2e6102848d76c762a7 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Tue, 4 Jul 2023 18:57:02 -0700
Subject: [PATCH] Break up long string literal

---
 bubblewrap.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/bubblewrap.c b/bubblewrap.c
index f8728c7..d834618 100644
--- a/bubblewrap.c
+++ b/bubblewrap.c
@@ -340,6 +340,8 @@ usage (int ecode, FILE *out)
            "    --dev-bind-try SRC DEST      Equal to --dev-bind but ignores non-existent SRC\n"
            "    --ro-bind SRC DEST           Bind mount the host path SRC readonly on DEST\n"
            "    --ro-bind-try SRC DEST       Equal to --ro-bind but ignores non-existent SRC\n"
+           );
+  fprintf (out,
            "    --bind-fd FD DEST            Bind open directory or path fd on DEST\n"
            "    --ro-bind-fd FD DEST         Bind open directory or path fd read-only on DEST\n"
            "    --remount-ro DEST            Remount DEST as readonly; does not recursively remount\n"
-- 
2.44.0

