From 65a767fb096d5cc125c2981dfbc298458091567f Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sat, 1 May 2021 01:39:26 -0700
Subject: [PATCH] Use __bswap32 instead of __builtin_bswap32

---
 tools/zynqmpbif.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/zynqmpbif.c b/tools/zynqmpbif.c
index 4d529ed785..233464cb47 100644
--- a/tools/zynqmpbif.c
+++ b/tools/zynqmpbif.c
@@ -517,7 +517,7 @@ static int bif_add_bit(struct bif_entry *bf)
 	debug("Bitstream Length: 0x%x\n", bitlen);
 	for (i = 0; i < bitlen; i += sizeof(uint32_t)) {
 		uint32_t *bitbin32 = (uint32_t *)&bitbin[i];
-		*bitbin32 = __builtin_bswap32(*bitbin32);
+		*bitbin32 = __bswap32(*bitbin32);
 	}
 
 	if (!bf->dest_dev)
-- 
2.31.1

