From 438abb276c7568391d757c7b3338e137307cdad5 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Fri, 30 Apr 2021 18:45:18 -0700
Subject: [PATCH] Add portable fallback for ctz, clz, and clzll

---
 include/common/attributes.h | 27 +++++++++++++++++++++++++--
 1 file changed, 25 insertions(+), 2 deletions(-)

diff --git a/include/common/attributes.h b/include/common/attributes.h
index 15892b9..47fed92 100644
--- a/include/common/attributes.h
+++ b/include/common/attributes.h
@@ -32,6 +32,7 @@
 
 #include <stddef.h>
 #include <assert.h>
+#include <strings.h>
 
 #ifdef __GNUC__
 #define ATTR_ALIAS __attribute__((may_alias))
@@ -146,7 +147,7 @@ static inline int clzll(const unsigned long long mask) {
         return clz((unsigned)mask) + 32;
 }
 #endif /* _WIN64 */
-#else /* !_MSC_VER */
+#elif defined(__GNUC__)
 static inline int ctz(const unsigned int mask) {
     return __builtin_ctz(mask);
 }
@@ -158,7 +159,29 @@ static inline int clz(const unsigned int mask) {
 static inline int clzll(const unsigned long long mask) {
     return __builtin_clzll(mask);
 }
-#endif /* !_MSC_VER */
+#else /* __GNUC__ */
+static inline int ctz(const unsigned int mask) {
+    return ffs(mask) - 1;
+}
+
+static inline int clz(unsigned int mask) {
+    mask >>= 1;
+    mask |= mask >> 1;
+    mask |= mask >> 2;
+    mask |= mask >> 4;
+    mask |= mask >> 8;
+    mask |= mask >> 16;
+    mask++;
+    return 32 - ffs(mask);
+}
+
+static inline int clzll(unsigned long long mask) {
+    if (mask >> 32)
+        return clz((unsigned)(mask >> 32));
+    else
+        return clz((unsigned)mask) + 32;
+}
+#endif /* !_MSC_VER && !__GNUC__ */
 
 #ifndef __has_feature
 #define __has_feature(x) 0
-- 
2.32.0

