From 78a6d851168c533469e8e22f210ba1654d377a78 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sun, 15 Oct 2023 08:33:50 -0600
Subject: [PATCH] Reuse dynbuf symbols for curl tool

---
 lib/dynbuf.h | 26 ++++++++++++--------------
 1 file changed, 12 insertions(+), 14 deletions(-)

diff --git a/lib/dynbuf.h b/lib/dynbuf.h
index 31a9130..507f5f5 100644
--- a/lib/dynbuf.h
+++ b/lib/dynbuf.h
@@ -26,23 +26,21 @@
 
 #include <curl/curl.h>
 
-#ifndef BUILDING_LIBCURL
 /* this renames the functions so that the tool code can use the same code
    without getting symbol collisions */
-#define Curl_dyn_init(a,b) curlx_dyn_init(a,b)
-#define Curl_dyn_add(a,b) curlx_dyn_add(a,b)
-#define Curl_dyn_addn(a,b,c) curlx_dyn_addn(a,b,c)
-#define Curl_dyn_addf curlx_dyn_addf
-#define Curl_dyn_vaddf curlx_dyn_vaddf
-#define Curl_dyn_free(a) curlx_dyn_free(a)
-#define Curl_dyn_ptr(a) curlx_dyn_ptr(a)
-#define Curl_dyn_uptr(a) curlx_dyn_uptr(a)
-#define Curl_dyn_len(a) curlx_dyn_len(a)
-#define Curl_dyn_reset(a) curlx_dyn_reset(a)
-#define Curl_dyn_tail(a,b) curlx_dyn_tail(a,b)
-#define Curl_dyn_setlen(a,b) curlx_dyn_setlen(a,b)
+#define curlx_dyn_init(a,b) Curl_dyn_init(a,b)
+#define curlx_dyn_add(a,b) Curl_dyn_add(a,b)
+#define curlx_dyn_addn(a,b,c) Curl_dyn_addn(a,b,c)
+#define curlx_dyn_addf Curl_dyn_addf
+#define curlx_dyn_vaddf Curl_dyn_vaddf
+#define curlx_dyn_free(a) Curl_dyn_free(a)
+#define curlx_dyn_ptr(a) Curl_dyn_ptr(a)
+#define curlx_dyn_uptr(a) Curl_dyn_uptr(a)
+#define curlx_dyn_len(a) Curl_dyn_len(a)
+#define curlx_dyn_reset(a) Curl_dyn_reset(a)
+#define curlx_dyn_tail(a,b) Curl_dyn_tail(a,b)
+#define curlx_dyn_setlen(a,b) Curl_dyn_setlen(a,b)
 #define curlx_dynbuf dynbuf /* for the struct name */
-#endif
 
 struct dynbuf {
   char *bufr;    /* point to a null-terminated allocated buffer */
-- 
2.42.0

