From 0c83f671f4f2a1b556a2f6a0cfd664cd13b077e0 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Fri, 16 Oct 2020 19:09:50 -0700
Subject: [PATCH] Reuse dynbuf symbols for curl tool

---
 lib/dynbuf.h | 26 ++++++++++++--------------
 1 file changed, 12 insertions(+), 14 deletions(-)

diff --git a/lib/dynbuf.h b/lib/dynbuf.h
index 31a913019..deabeb0e6 100644
--- a/lib/dynbuf.h
+++ b/lib/dynbuf.h
@@ -26,23 +26,21 @@
 
 #include <curl/curl.h>
 
-#ifndef BUILDING_LIBCURL
 /* this renames the functions so that the tool code can use the same code
    without getting symbol collisions */
-#define Curl_dyn_init(a,b) curlx_dyn_init(a,b)
-#define Curl_dyn_add(a,b) curlx_dyn_add(a,b)
-#define Curl_dyn_addn(a,b,c) curlx_dyn_addn(a,b,c)
-#define Curl_dyn_addf curlx_dyn_addf
-#define Curl_dyn_vaddf curlx_dyn_vaddf
-#define Curl_dyn_free(a) curlx_dyn_free(a)
-#define Curl_dyn_ptr(a) curlx_dyn_ptr(a)
-#define Curl_dyn_uptr(a) curlx_dyn_uptr(a)
-#define Curl_dyn_len(a) curlx_dyn_len(a)
-#define Curl_dyn_reset(a) curlx_dyn_reset(a)
-#define Curl_dyn_tail(a,b) curlx_dyn_tail(a,b)
-#define Curl_dyn_setlen(a,b) curlx_dyn_setlen(a,b)
+#define curlx_dyn_init Curl_dyn_init
+#define curlx_dyn_add Curl_dyn_add
+#define curlx_dyn_addn Curl_dyn_addn
+#define curlx_dyn_addf Curl_dyn_addf
+#define curlx_dyn_vaddf Curl_dyn_vaddf
+#define curlx_dyn_free Curl_dyn_free
+#define curlx_dyn_ptr Curl_dyn_ptr
+#define curlx_dyn_uptr Curl_dyn_uptr
+#define curlx_dyn_len Curl_dyn_len
+#define curlx_dyn_reset Curl_dyn_reset
+#define curlx_dyn_tail Curl_dyn_tail
+#define curlx_dyn_setlen Curl_dyn_setlen
 #define curlx_dynbuf dynbuf /* for the struct name */
-#endif
 
 struct dynbuf {
   char *bufr;    /* point to a null-terminated allocated buffer */
-- 
2.42.0

