From 4a4a14e99d8bdf4f25bbf60ed4f1a62f19ed3a57 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Mon, 6 Sep 2021 18:19:30 -0700
Subject: [PATCH] Use double instead of long double for progress calculation

---
 lib/progress.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/progress.c b/lib/progress.c
index 4bcd615eb..d46a3a06d 100644
--- a/lib/progress.c
+++ b/lib/progress.c
@@ -377,7 +377,7 @@ static curl_off_t trspeed(curl_off_t size, /* number of bytes */
 {
   if(us < 1)
     return size * 1000000;
-  return (curl_off_t)((long double)size/us * 1000000);
+  return (curl_off_t)((double)size/us * 1000000);
 }
 
 /* returns TRUE if it's time to show the progress meter */
-- 
2.32.0

