From 299d37f57b16a4613fabd179288039e883c161cb Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Fri, 11 Aug 2017 23:08:10 -0700
Subject: [PATCH] Set CTLSOCKET to /run/ntpd.sock

---
 src/ntpd.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ntpd.h b/src/ntpd.h
index 39b8e7b..41a0b22 100644
--- a/src/ntpd.h
+++ b/src/ntpd.h
@@ -48,7 +48,7 @@
 #define	LOCALSTATEDIR	"/var"
 #endif
 #define	DRIFTFILE	LOCALSTATEDIR "/db/ntpd.drift"
-#define	CTLSOCKET	LOCALSTATEDIR "/run/ntpd.sock"
+#define	CTLSOCKET	"/run/ntpd.sock"
 
 #if defined(SO_SETFIB)
 #define	SO_RTABLE	SO_SETFIB
-- 
2.14.1

