From 2760ceff5d4817963fb7882bfaf8852c8dc4460c Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Fri, 11 Aug 2017 23:08:10 -0700
Subject: [PATCH] Set CTLSOCKET to /run/ntpd.sock

---
 src/ntpd.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ntpd.h b/src/ntpd.h
index ced81b9ff..7364c5aa8 100644
--- a/src/ntpd.h
+++ b/src/ntpd.h
@@ -49,7 +49,7 @@
 #define	LOCALSTATEDIR	"/var"
 #endif
 #define	DRIFTFILE	LOCALSTATEDIR "/db/ntpd.drift"
-#define	CTLSOCKET	LOCALSTATEDIR "/run/ntpd.sock"
+#define	CTLSOCKET	"/run/ntpd.sock"
 
 #define	INTERVAL_QUERY_NORMAL		30	/* sync to peers every n secs */
 #define	INTERVAL_QUERY_PATHETIC		60
-- 
2.29.2

