From dfb7c96e207a1c6d298d417c2e0b8fd79821d8d3 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sat, 27 Jun 2020 01:09:58 -0700
Subject: [PATCH] Enable add.interactive.usebuiltin by default

---
 builtin/add.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/builtin/add.c b/builtin/add.c
index ef6b619c45..c2b326abfd 100644
--- a/builtin/add.c
+++ b/builtin/add.c
@@ -239,6 +239,7 @@ int run_add_interactive(const char *revision, const char *patch_mode,
 
 	if (use_builtin_add_i < 0) {
 		int experimental;
+		use_builtin_add_i = 1;
 		if (!git_config_get_bool("add.interactive.usebuiltin",
 					 &use_builtin_add_i))
 			; /* ok */
-- 
2.32.0

