From 998329082e2fa9452c817fe6be6bcc908c13e8a7 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sat, 27 Jun 2020 01:09:58 -0700
Subject: [PATCH] Enable add.interactive.usebuiltin by default

---
 builtin/add.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/builtin/add.c b/builtin/add.c
index a825887c50..737ef23492 100644
--- a/builtin/add.c
+++ b/builtin/add.c
@@ -194,6 +194,7 @@ int run_add_interactive(const char *revision, const char *patch_mode,
 
 	if (use_builtin_add_i < 0) {
 		int experimental;
+		use_builtin_add_i = 1;
 		if (!git_config_get_bool("add.interactive.usebuiltin",
 					 &use_builtin_add_i))
 			; /* ok */
-- 
2.28.0

