From e85b8d3ae6b808e48a1ffd530cf51fee083c0766 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sat, 27 Jun 2020 01:09:58 -0700
Subject: [PATCH] Enable add.interactive.usebuiltin by default

---
 builtin/add.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/builtin/add.c b/builtin/add.c
index 298e0114f9..565fdab636 100644
--- a/builtin/add.c
+++ b/builtin/add.c
@@ -192,9 +192,11 @@ int run_add_interactive(const char *revision, const char *patch_mode,
 	int use_builtin_add_i =
 		git_env_bool("GIT_TEST_ADD_I_USE_BUILTIN", -1);
 
-	if (use_builtin_add_i < 0)
+	if (use_builtin_add_i < 0) {
+		use_builtin_add_i = 1;
 		git_config_get_bool("add.interactive.usebuiltin",
 				    &use_builtin_add_i);
+	}
 
 	if (use_builtin_add_i == 1) {
 		enum add_p_mode mode;
-- 
2.27.0

