From 76d023b885150b65ceeee179883fdf97bd9b8012 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sun, 16 Aug 2020 17:15:34 -0700
Subject: [PATCH] Use flexible array member instead of zero-length array

---
 setpci.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setpci.c b/setpci.c
index 90ca726..ead2bb9 100644
--- a/setpci.c
+++ b/setpci.c
@@ -37,7 +37,7 @@ struct op {
   unsigned int width;			/* Byte width of the access */
   unsigned int num_values;		/* Number of values to write; 0=read */
   unsigned int number;                 /* The n-th capability of that id */
-  struct value values[0];
+  struct value values[];
 };
 
 struct group {
-- 
2.28.0

