From 9e6d578db0a4a55be5dbba40aba78e830181734d Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sun, 16 Aug 2020 17:16:31 -0700
Subject: [PATCH] Use strerror instead of GNU %m format specifier

---
 ls-kernel.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/ls-kernel.c b/ls-kernel.c
index 13b29f7..1f7933a 100644
--- a/ls-kernel.c
+++ b/ls-kernel.c
@@ -8,6 +8,7 @@
  *	SPDX-License-Identifier: GPL-2.0-or-later
  */
 
+#include <errno.h>
 #include <stdio.h>
 #include <string.h>
 
@@ -125,12 +126,12 @@ show_kernel_init(void)
     {
       f = fopen(name, "r");
       if (!f)
-	die("Cannot open pcimap file %s: %m", name);
+	die("Cannot open pcimap file %s: %s", name, strerror(errno));
     }
   else
     {
       if (uname(&uts) < 0)
-	die("uname() failed: %m");
+	die("uname() failed: %s", strerror(errno));
       name = alloca(64 + strlen(uts.release));
       sprintf(name, "/lib/modules/%s/modules.pcimap", uts.release);
       f = fopen(name, "r");
-- 
2.49.0

