From 3c881bf9f7c67b36ada5b201e6e860d1e9dac52b Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sat, 14 Oct 2017 01:00:22 -0700
Subject: [PATCH 1/2] Remove quotes from @dbfile@ in man pages

---
 nsd.8.in      | 4 ++--
 nsd.conf.5.in | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/nsd.8.in b/nsd.8.in
index 5e488958..babaa06a 100644
--- a/nsd.8.in
+++ b/nsd.8.in
@@ -115,7 +115,7 @@ Do not fork, stay in the foreground.
 Use the specified
 .I database
 instead of the default of
-.IR '@dbfile@' .
+.IR @dbfile@ .
 If a 
 .B zonesdir: 
 is specified in the config file this path can be relative to that 
@@ -224,7 +224,7 @@ SIGUSR1
 Dump BIND8\-style statistics into the log. Ignored otherwise.
 .SH "FILES"
 .TP
-"@dbfile@"
+@dbfile@
 default
 .B NSD
 database
diff --git a/nsd.conf.5.in b/nsd.conf.5.in
index 74e7c6f8..7825d4f8 100644
--- a/nsd.conf.5.in
+++ b/nsd.conf.5.in
@@ -201,7 +201,7 @@ If yes, NSD listens to IPv6 connections.  Default yes.
 .TP
 .B database:\fR <filename>
 By default 
-.I '@dbfile@'
+.I @dbfile@
 is used. The specified file is used to store the compiled 
 zone information. Same as commandline option 
 .BR \-f.
@@ -876,7 +876,7 @@ also function as a resolver or cache. The configuration options that
 BIND9 has for the resolver or caching thus have no equivalents for NSD.
 .SH "FILES"
 .TP
-"@dbfile@"
+@dbfile@
 default
 .B NSD
 database
-- 
2.18.0

