From 8dee43a5ff31bbe2b463ca66b168202501a14a92 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sun, 12 Jun 2016 20:51:43 -0700
Subject: [PATCH] mount: Don't pass {,no}auto to kernel

---
 mount.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mount.c b/mount.c
index 2eb175c..4ce6680 100644
--- a/mount.c
+++ b/mount.c
@@ -29,6 +29,7 @@ struct {
 	{ "dirsync",    NULL,           MS_DIRSYNC     },
 	{ "nodev",      "dev",          MS_NODEV       },
 	{ "noatime",    "atime",        MS_NOATIME     },
+	{ "noauto",     "auto",         0              },
 	{ "nodiratime", "diratime",     MS_NODIRATIME  },
 	{ "noexec",     "exec",         MS_NOEXEC      },
 	{ "nosuid",     "suid",         MS_NOSUID      },
-- 
2.8.1

