From d575dcbe5b779829881974623d2585481e71c38e Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sun, 16 Jun 2019 12:39:04 -0700
Subject: [PATCH] Remove semicolon after function definitions

---
 lib/json_print.c | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/lib/json_print.c b/lib/json_print.c
index 8e7f32dc..8929e4d8 100644
--- a/lib/json_print.c
+++ b/lib/json_print.c
@@ -138,15 +138,15 @@ void close_json_array(enum output_type type, const char *str)
 			color_fprintf(stdout, color, fmt, value);          \
 		}							\
 	}
-_PRINT_FUNC(int, int);
-_PRINT_FUNC(s64, int64_t);
-_PRINT_FUNC(hhu, unsigned char);
-_PRINT_FUNC(hu, unsigned short);
-_PRINT_FUNC(uint, unsigned int);
-_PRINT_FUNC(u64, uint64_t);
-_PRINT_FUNC(luint, unsigned long);
-_PRINT_FUNC(lluint, unsigned long long);
-_PRINT_FUNC(float, double);
+_PRINT_FUNC(int, int)
+_PRINT_FUNC(s64, int64_t)
+_PRINT_FUNC(hhu, unsigned char)
+_PRINT_FUNC(hu, unsigned short)
+_PRINT_FUNC(uint, unsigned int)
+_PRINT_FUNC(u64, uint64_t)
+_PRINT_FUNC(luint, unsigned long)
+_PRINT_FUNC(lluint, unsigned long long)
+_PRINT_FUNC(float, double)
 #undef _PRINT_FUNC
 
 #define _PRINT_NAME_VALUE_FUNC(type_name, type, format_char)		 \
@@ -158,8 +158,8 @@ _PRINT_FUNC(float, double);
 			 "%s %%"#format_char, name);			 \
 		print_##type_name(PRINT_ANY, name, format, value);	 \
 	}
-_PRINT_NAME_VALUE_FUNC(uint, unsigned int, u);
-_PRINT_NAME_VALUE_FUNC(string, const char*, s);
+_PRINT_NAME_VALUE_FUNC(uint, unsigned int, u)
+_PRINT_NAME_VALUE_FUNC(string, const char*, s)
 #undef _PRINT_NAME_VALUE_FUNC
 
 void print_color_string(enum output_type type,
-- 
2.25.0

