From 8138dce28e676e77b189f3bd94d640ea268d361f Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sat, 4 Jun 2016 23:42:30 -0700
Subject: [PATCH] install: Handle -c flag

---
 xinstall.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/xinstall.c b/xinstall.c
index f5f6956..bf921fb 100644
--- a/xinstall.c
+++ b/xinstall.c
@@ -156,6 +156,9 @@ main(int argc, char *argv[])
 	char *p;
 
 	ARGBEGIN {
+	case 'c':
+		/* no-op for compatibility */
+		break;
 	case 'd':
 		dflag = 1;
 		break;
-- 
2.8.1

