From 2083459cade962e76d58a65e6c211e3aa9cf6d64 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Wed, 7 Dec 2016 23:41:29 -0800
Subject: [PATCH] concat: Use fixed buffer size (larger than BUFSIZ on musl)

---
 libutil/concat.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libutil/concat.c b/libutil/concat.c
index 2e9aa52..461a4cc 100644
--- a/libutil/concat.c
+++ b/libutil/concat.c
@@ -6,7 +6,7 @@
 int
 concat(int f1, const char *s1, int f2, const char *s2)
 {
-	char buf[BUFSIZ];
+	char buf[8192];
 	ssize_t n;
 
 	while ((n = read(f1, buf, sizeof(buf))) > 0) {
-- 
2.11.0

