From e96eee39316e30dfdf965e4894fc77a8bf43e40f Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Fri, 15 Jul 2016 19:21:28 -0700
Subject: [PATCH] Fix poll.h include

---
 lib/mount.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/mount.c b/lib/mount.c
index 2150189..7a18c11 100644
--- a/lib/mount.c
+++ b/lib/mount.c
@@ -21,7 +21,7 @@
 #include <string.h>
 #include <fcntl.h>
 #include <errno.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/socket.h>
 #include <sys/un.h>
 #include <sys/wait.h>
-- 
2.13.2

