From 8926abd905339e40bf096c904d900caa42e3262f Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Fri, 15 Jul 2016 19:21:28 -0700
Subject: [PATCH] Fix poll.h include

---
 lib/mount.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/mount.c b/lib/mount.c
index 227a408..0fc7d15 100644
--- a/lib/mount.c
+++ b/lib/mount.c
@@ -20,7 +20,7 @@
 #include <string.h>
 #include <fcntl.h>
 #include <errno.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/socket.h>
 #include <sys/un.h>
 #include <sys/wait.h>
-- 
2.9.1

