From 3ebf1956536d0becda1267bb465010d919bca0ed Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sun, 9 Jul 2017 02:35:33 -0700
Subject: [PATCH] Only enable FUSE_SYMVER on glibc

---
 lib/fuse_misc.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/fuse_misc.h b/lib/fuse_misc.h
index 2f6663e..c95ca57 100644
--- a/lib/fuse_misc.h
+++ b/lib/fuse_misc.h
@@ -13,7 +13,7 @@
     - confuse the dynamic linker in uClibc
     - not supported on MacOSX (in MachO binary format)
 */
-#if (!defined(__UCLIBC__) && !defined(__APPLE__))
+#ifdef __GLIBC__
 #define FUSE_SYMVER(x) __asm__(x)
 #else
 #define FUSE_SYMVER(x)
-- 
2.14.1

