From 9e9c7141aceb6e144bd132fa44f605213125d15a Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sun, 9 Jul 2017 02:35:45 -0700
Subject: [PATCH] Disable iconv module

---
 lib/fuse.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/lib/fuse.c b/lib/fuse.c
index 75f13d7..42b0cec 100644
--- a/lib/fuse.c
+++ b/lib/fuse.c
@@ -196,9 +196,8 @@ struct fuse_context_i {
 	fuse_req_t req;
 };
 
-/* Defined by FUSE_REGISTER_MODULE() in lib/modules/subdir.c and iconv.c.  */
+/* Defined by FUSE_REGISTER_MODULE() in lib/modules/subdir.c.  */
 extern fuse_module_factory_t fuse_module_subdir_factory;
-extern fuse_module_factory_t fuse_module_iconv_factory;
 
 static pthread_key_t fuse_context_key;
 static pthread_mutex_t fuse_context_lock = PTHREAD_MUTEX_INITIALIZER;
@@ -4475,7 +4474,6 @@ void fuse_lib_help(struct fuse_args *args)
 
 	/* Print help for builtin modules */
 	print_module_help("subdir", &fuse_module_subdir_factory);
-	print_module_help("iconv", &fuse_module_iconv_factory);
 
 	/* Parse command line options in case we need to
 	   activate more modules */
@@ -4662,7 +4660,6 @@ struct fuse *fuse_new_31(struct fuse_args *args,
 	if (builtin_modules_registered == 0) {
 		/* If not, register them. */
 		fuse_register_module("subdir", fuse_module_subdir_factory, NULL);
-		fuse_register_module("iconv", fuse_module_iconv_factory, NULL);
 		builtin_modules_registered= 1;
 	}
 	pthread_mutex_unlock(&fuse_context_lock);
-- 
2.14.1

