From 9f5994077aed134d284c168b61c3167ac9837c98 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sun, 11 Aug 2019 03:41:23 +0000
Subject: [PATCH] Avoid statement expressions for get_aligned_le*

---
 src/utils/platform.h | 15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

diff --git a/src/utils/platform.h b/src/utils/platform.h
index b2ad856..a76c87f 100644
--- a/src/utils/platform.h
+++ b/src/utils/platform.h
@@ -12,7 +12,18 @@
 #include "includes.h"
 #include "common.h"
 
-#define get_unaligned_le16(p)	WPA_GET_LE16((void *) (p))
-#define get_unaligned_le32(p)	WPA_GET_LE32((void *) (p))
+static inline u16 get_unaligned_le16(void *p)
+{
+	u16 v;
+	memcpy(&v, p, sizeof(v));
+	return le_to_host16(v);
+}
+
+static inline u32 get_unaligned_le32(void *p)
+{
+	u32 v;
+	memcpy(&v, p, sizeof(v));
+	return le_to_host32(v);
+}
 
 #endif /* PLATFORM_H */
-- 
2.45.1

