From b7f437785aa3616066e5c5845e466f49b50db263 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Thu, 22 Mar 2018 23:42:12 -0700
Subject: [PATCH] Fix build with libressl-2.7.0

---
 Modules/_ssl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Modules/_ssl.c b/Modules/_ssl.c
index c54e43c2b4..93e15c91d0 100644
--- a/Modules/_ssl.c
+++ b/Modules/_ssl.c
@@ -104,7 +104,7 @@ struct py_ssl_library_code {
 /* Include generated data (error codes) */
 #include "_ssl_data.h"
 
-#if (OPENSSL_VERSION_NUMBER >= 0x10100000L) && !defined(LIBRESSL_VERSION_NUMBER)
+#if (OPENSSL_VERSION_NUMBER >= 0x10100000L) && (!defined(LIBRESSL_VERSION_NUMBER) || LIBRESSL_VERSION_NUMBER >= 0x2070000fL)
 #  define OPENSSL_VERSION_1_1 1
 #endif
 
-- 
2.17.0

