From ce8a3ec6f269d0d44ecf83669b5567ca9c934bc4 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Thu, 22 Mar 2018 23:42:12 -0700
Subject: [PATCH] Fix build with libressl-2.7.0

---
 Modules/_ssl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Modules/_ssl.c b/Modules/_ssl.c
index df8c6a7d96..0e18cd887f 100644
--- a/Modules/_ssl.c
+++ b/Modules/_ssl.c
@@ -99,7 +99,7 @@ struct py_ssl_library_code {
 /* Include generated data (error codes) */
 #include "_ssl_data.h"
 
-#if (OPENSSL_VERSION_NUMBER >= 0x10100000L) && !defined(LIBRESSL_VERSION_NUMBER)
+#if (OPENSSL_VERSION_NUMBER >= 0x10100000L) && (!defined(LIBRESSL_VERSION_NUMBER) || LIBRESSL_VERSION_NUMBER >= 0x2070000fL)
 #  define OPENSSL_VERSION_1_1 1
 #endif
 
-- 
2.16.3

