From f7bd663aa19c84da21d5f6c15ca50daa9862ad9a Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sun, 21 Apr 2019 12:55:53 -0700
Subject: [PATCH] load: Error out when there are too many phi args

---
 load.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/load.c b/load.c
index 8e2dd64..9894000 100644
--- a/load.c
+++ b/load.c
@@ -330,6 +330,8 @@ def(Slice sl, bits msk, Blk *b, Ins *i, Loc *il)
 	p->to = r;
 	p->cls = sl.cls;
 	p->narg = b->npred;
+	if (b->npred >= NPred)
+		die("def, too many phi args (%u)", b->npred);
 	for (np=0; np<b->npred; ++np) {
 		bp = b->pred[np];
 		if (!bp->s2
-- 
2.21.0

