From 294fedc93dbeac68f0beec1eeea62be30227b025 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Fri, 31 May 2019 13:31:04 -0700
Subject: [PATCH] increase NString to 72

---
 all.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/all.h b/all.h
index 37980d3..f49b4ef 100644
--- a/all.h
+++ b/all.h
@@ -31,7 +31,7 @@ typedef struct Dat Dat;
 typedef struct Target Target;
 
 enum {
-	NString = 64,
+	NString = 72,
 	NIns    = 1 << 20,
 	NAlign  = 3,
 	NField  = 32,
-- 
2.32.0

