From 44da6aa9a8a91c115c22458bc7bc97d0248beaeb Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sat, 6 Jul 2019 01:18:59 -0700
Subject: [PATCH] Only use GCC pragmas with gcc

---
 src/ks_tables.h | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/src/ks_tables.h b/src/ks_tables.h
index a6db8de..a9f30d9 100644
--- a/src/ks_tables.h
+++ b/src/ks_tables.h
@@ -5,8 +5,10 @@
  * https://raw.github.com/xkbcommon/libxkbcommon/master/src/ks_tables.h
  */
 
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Woverlength-strings"
+#ifdef __GNUC__
+# pragma GCC diagnostic push
+# pragma GCC diagnostic ignored "-Woverlength-strings"
+#endif
 static const char *keysym_names =
     "0\0"
     "1\0"
@@ -2417,7 +2419,9 @@ static const char *keysym_names =
     "Zstroke\0"
     "zstroke\0"
 ;
-#pragma GCC diagnostic pop
+#ifdef __GNUC__
+# pragma GCC diagnostic pop
+#endif
 
 struct name_keysym {
     xkb_keysym_t keysym;
-- 
2.22.0

